// @salutejs/plasma-core/components/Card/CardTypography.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardParagraph2 = exports.CardParagraph1 = exports.CardBody2 = exports.CardBody1 = exports.CardFootnote2 = exports.CardFootnote1 = exports.CardHeadline3 = exports.CardHeadline2 = exports.CardHeadline1 = void 0;

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _tokens = /*#__PURE__*/require("../../tokens");

var _Typography = /*#__PURE__*/require("../Typography");

var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/* stylelint-disable */
// TODO:
(0, _collectPackageInfo.collectPackageInfo)('Card/CardTypography');
var views = {
  primary: _tokens.primary,
  secondary: _tokens.secondary,
  tertiary: _tokens.tertiary
};

var linesMixin = function linesMixin(_ref) {
  var _ref$lines = _ref.lines,
      lines = _ref$lines === void 0 ? 2 : _ref$lines;
  return (0, _styledComponents.css)(["display:-webkit-box;overflow:hidden;box-sizing:border-box;white-space:normal;-webkit-box-orient:vertical;-webkit-line-clamp:", ";"], lines);
};

var viewMixin = function viewMixin(_ref2) {
  var _ref2$view = _ref2.view,
      view = _ref2$view === void 0 ? 'primary' : _ref2$view;
  return (0, _styledComponents.css)(["transition:", ";color:", ";"], function (_ref3) {
    var theme = _ref3.theme;
    return theme.lowPerformance ? 'unset' : 'color 0.1s ease-in-out';
  }, views[view]);
};

var CardHeadline1 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Headline1).withConfig({
  componentId: "plasma__sc-zyrk75-0"
})(["", ";"], viewMixin);
exports.CardHeadline1 = CardHeadline1;
var CardHeadline2 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Headline2).withConfig({
  componentId: "plasma__sc-zyrk75-1"
})(["", ";"], viewMixin);
exports.CardHeadline2 = CardHeadline2;
var CardHeadline3 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Headline3).withConfig({
  componentId: "plasma__sc-zyrk75-2"
})(["", ";"], viewMixin);
exports.CardHeadline3 = CardHeadline3;
var CardFootnote1 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote1).withConfig({
  componentId: "plasma__sc-zyrk75-3"
})(["", ";"], viewMixin);
exports.CardFootnote1 = CardFootnote1;
var CardFootnote2 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote2).withConfig({
  componentId: "plasma__sc-zyrk75-4"
})(["", ";"], viewMixin);
exports.CardFootnote2 = CardFootnote2;
var CardBody1 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body1).withConfig({
  componentId: "plasma__sc-zyrk75-5"
})(["", ";", ";"], linesMixin, viewMixin);
exports.CardBody1 = CardBody1;
var CardBody2 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body2).withConfig({
  componentId: "plasma__sc-zyrk75-6"
})(["", ";", ";"], linesMixin, viewMixin);
exports.CardBody2 = CardBody2;
var CardParagraph1 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.ParagraphText1).withConfig({
  componentId: "plasma__sc-zyrk75-7"
})(["", ";", ";"], linesMixin, viewMixin);
exports.CardParagraph1 = CardParagraph1;
var CardParagraph2 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.ParagraphText2).withConfig({
  componentId: "plasma__sc-zyrk75-8"
})(["", ";", ";"], linesMixin, viewMixin);
exports.CardParagraph2 = CardParagraph2;
