// @salutejs/plasma-tokens/es/brands/sbermarket/values.js
// Generated by robots, do not change this manually!
/** Основной цвет текста */
export var textPrimary = '#F2F5F7';
/** Вторичный цвет текста */
export var textSecondary = '#A8B4BD';
/** Третичный цвет текста */
export var textTertiary = '#7D868C';
/** Сплошной наборный текст */
export var textParagraph = '#BEC9D1';
/** Акцентный цвет */
export var textAccent = '#24B23E';
/** Цвет ошибки */
export var textNegative = '#FF2E43';
/** Дополнительный акцент */
export var textAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
export var textAccentGold = '#FFD83D';
/** Акцентный цвет бренда */
export var textAccentB2B = '#04C6C9';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'rgba(242, 245, 247, 0.96)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'rgba(242, 245, 247, 0.64)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'rgba(242, 245, 247, 0.4)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'rgba(242, 245, 247, 0.8)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = '#24B23E';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = '#FF2E43';
/** Дополнительный акцент */
export var onDarkTextAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
export var onDarkTextAccentGold = '#FFD83D';
/** Акцентный цвет бренда */
export var onDarkTextAccentB2B = '#04C6C9';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = '#24282B';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = '#8C959C';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = '#BEC9D1';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = '#24282B';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = '#108E26';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = '#F31B31';
/** Дополнительный акцент */
export var onLightTextAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
export var onLightTextAccentGold = '#FFD83D';
/** Акцентный цвет бренда */
export var onLightTextAccentB2B = '#0C8688';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = '#24282B';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = '#24282B';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = '#31373B';
/** Основной фон для карточек */
export var surfaceSolidCard = '#15181A';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = '#24B23E';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = '#FF2E43';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentB2B = '#0DA8AB';
/** Дополнительный акцент для обозначения акций */
export var surfaceAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
export var surfaceAccentGold = '#FFD83D';
/** Дополнительный акцент */
export var surfaceAccentBlue = '#5993FF';
/** Пунктирные дивайдеры и неактивные контролы */
export var surfaceAdditional = '#494F54';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondary = '#24282B';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = '#31373B';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = '#15181A';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = '#24B23E';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradient = 'linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = '#FF2E43';
/** Акцентный фон поверхности/контрола */
export var onDarkSurfaceAccentB2B = '#0DA8AB';
/** Дополнительный акцент для обозначения акций */
export var onDarkSurfaceAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
export var onDarkSurfaceAccentGold = '#FFD83D';
/** Дополнительный акцент */
export var onDarkSurfaceAccentBlue = '#5993FF';
/** Пунктирные дивайдеры и неактивные контролы */
export var onDarkSurfaceAdditional = '#494F54';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondary = '#F2F5F7';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiary = '#E4EBF0';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = '#FFFFFF';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = '#24B23E';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradient = 'linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = '#F31B31';
/** Акцентный фон поверхности/контрола */
export var onLightSurfaceAccentB2B = '#0DA8AB';
/** Дополнительный акцент для обозначения акций */
export var onLightSurfaceAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
export var onLightSurfaceAccentGold = '#FFD83D';
/** Дополнительный акцент */
export var onLightSurfaceAccentBlue = '#3F81FD';
/** Пунктирные дивайдеры и неактивные контролы */
export var onLightSurfaceAdditional = '#D1DDE5';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerRed = '#2E090D';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerGold = '#1F1903';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerGreen = '#071F0C';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerMalachite = '#0A1C1C';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerBlue = '#0A162B';
/** Основной фон */
export var backgroundPrimary = '#08090A';
/** Вторичный фон */
export var backgroundOverlay = 'rgba(21, 24, 26, 0.56)';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'rgba(242, 245, 247, 0.96)';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'rgba(242, 245, 247, 0.64)';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'rgba(242, 245, 247, 0.4)';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = '#24282B';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = '#8C959C';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = '#BEC9D1';
/** @deprecated instead use textPrimary */
export var text = '#F2F5F7';
/** @deprecated instead use textPrimary */
export var primary = '#F2F5F7';
/** @deprecated instead use textSecondary */
export var secondary = '#A8B4BD';
/** @deprecated instead use textTertiary */
export var tertiary = '#7D868C';
/** @deprecated instead use textParagraph */
export var paragraph = '#BEC9D1';
/** @deprecated instead use backgroundPrimary */
export var background = '#08090A';
/** @deprecated instead use textAccent */
export var accent = '#24B23E';
/** @deprecated instead use textNegative */
export var critical = '#FF2E43';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = '#24282B';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = '#31373B';
/** @deprecated instead use textAccent */
export var buttonAccent = '#24B23E';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = '#FF2E43';
