// @salutejs/plasma-ui/components/Header/HeaderRoot.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HeaderRoot = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _color = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("color"));

var _utils = /*#__PURE__*/require("../../utils");

var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");

var _excluded = ["children", "size", "gradientColor"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

(0, _collectPackageInfo.collectPackageInfo)('Header/HeaderRoot');
var sizes = {
  sberBox: /*#__PURE__*/(0, _styledComponents.css)(["--plasma-header-height:2.25rem;--plasma-header-pt:1.875rem;--plasma-header-pb:0.875rem;"]),
  sberPortal: /*#__PURE__*/(0, _styledComponents.css)(["--plasma-header-height:2.25rem;--plasma-header-pt:1.625rem;--plasma-header-pb:0.625rem;"]),
  mobile: /*#__PURE__*/(0, _styledComponents.css)(["--plasma-header-height:2.25rem;--plasma-header-pt:0.375rem;--plasma-header-pb:0.375rem;"])
};

// TODO: https://github.com/salute-developers/plasma/issues/232
var StyledHeaderRoot = /*#__PURE__*/_styledComponents["default"].header.withConfig({
  componentId: "plasma-ui__sc-rhe7wm-0"
})(["box-sizing:content-box;width:100%;height:var(--plasma-header-height);padding-top:var(--plasma-header-pt);padding-bottom:var(--plasma-header-pb);", " ", ""], function (_ref) {
  var $size = _ref.$size,
      theme = _ref.theme;
  return $size ? sizes[$size] // Выберет указанный размер или разложит размеры по брейкпоинтам
  : (0, _styledComponents.css)(["", " ", " ", " ", ""], (0, _utils.mediaQuery)('S', theme.deviceScale)(sizes.mobile), (0, _utils.mediaQuery)('M', theme.deviceScale)(sizes.sberPortal), (0, _utils.mediaQuery)('L', theme.deviceScale)(sizes.sberBox), (0, _utils.mediaQuery)('XL', theme.deviceScale)(sizes.sberBox));
}, function (_ref2) {
  var $gradientColor = _ref2.$gradientColor;
  return $gradientColor && (0, _styledComponents.css)(["&::before{content:'';position:absolute;top:0;left:0;right:0;height:calc(var(--plasma-header-height) + var(--plasma-header-pt) + var(--plasma-header-pb));background-image:linear-gradient( 180deg,", ",", " );}"], $gradientColor, (0, _color["default"])($gradientColor).alpha(0).string());
});

var StyledInner = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-rhe7wm-1"
})(["position:relative;display:flex;align-items:center;flex-direction:row;width:100%;height:100%;"]);

/**
 * Корневой узел для шапки.
 */
var HeaderRoot = /*#__PURE__*/(0, _react.forwardRef)(function (_ref3, ref) {
  var children = _ref3.children,
      size = _ref3.size,
      gradientColor = _ref3.gradientColor,
      rest = _objectWithoutProperties(_ref3, _excluded);

  return /*#__PURE__*/_react["default"].createElement(StyledHeaderRoot, _extends({
    ref: ref
  }, rest, {
    $size: size,
    $gradientColor: gradientColor
  }), /*#__PURE__*/_react["default"].createElement(StyledInner, null, children));
});
exports.HeaderRoot = HeaderRoot;
