// @salutejs/plasma-typo/lib/cjs/archetypes/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sbermarketTypoProperties = exports.sbermarket = exports.soulmateTypoProperties = exports.soulmate = exports.sageTypoProperties = exports.sage = exports.rulerTypoProperties = exports.ruler = exports.plasmaTypoProperties = exports.plasma = exports.mageTypoProperties = exports.mage = void 0;
var mage_1 = require("./mage");
Object.defineProperty(exports, "mage", { enumerable: true, get: function () { return mage_1.mage; } });
Object.defineProperty(exports, "mageTypoProperties", { enumerable: true, get: function () { return mage_1.mageTypoProperties; } });
var plasma_1 = require("./plasma");
Object.defineProperty(exports, "plasma", { enumerable: true, get: function () { return plasma_1.plasma; } });
Object.defineProperty(exports, "plasmaTypoProperties", { enumerable: true, get: function () { return plasma_1.plasmaTypoProperties; } });
var ruler_1 = require("./ruler");
Object.defineProperty(exports, "ruler", { enumerable: true, get: function () { return ruler_1.ruler; } });
Object.defineProperty(exports, "rulerTypoProperties", { enumerable: true, get: function () { return ruler_1.rulerTypoProperties; } });
var sage_1 = require("./sage");
Object.defineProperty(exports, "sage", { enumerable: true, get: function () { return sage_1.sage; } });
Object.defineProperty(exports, "sageTypoProperties", { enumerable: true, get: function () { return sage_1.sageTypoProperties; } });
var soulmate_1 = require("./soulmate");
Object.defineProperty(exports, "soulmate", { enumerable: true, get: function () { return soulmate_1.soulmate; } });
Object.defineProperty(exports, "soulmateTypoProperties", { enumerable: true, get: function () { return soulmate_1.soulmateTypoProperties; } });
var sbermarket_1 = require("./sbermarket");
Object.defineProperty(exports, "sbermarket", { enumerable: true, get: function () { return sbermarket_1.sbermarket; } });
Object.defineProperty(exports, "sbermarketTypoProperties", { enumerable: true, get: function () { return sbermarket_1.sbermarketTypoProperties; } });
