// @salutejs/plasma-tokens/es/brands/sbermarket_metro/index.js
// Generated by robots, do not change this manually!
/** Основной цвет текста */
export var textPrimary = 'var(--text-primary, #F2F5F7)';
/** Вторичный цвет текста */
export var textSecondary = 'var(--text-secondary, #A8B4BD)';
/** Третичный цвет текста */
export var textTertiary = 'var(--text-tertiary, #7D868C)';
/** Сплошной наборный текст */
export var textParagraph = 'var(--text-paragraph, #BEC9D1)';
/** Акцентный цвет */
export var textAccent = 'var(--text-accent, #0064FE)';
/** Цвет ошибки */
export var textNegative = 'var(--text-negative, #FF2E43)';
/** Дополнительный акцент */
export var textAccentDiscount = 'var(--text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var textAccentGold = 'var(--text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
export var textAccentB2B = 'var(--text-accent-b2-b, #04C6C9)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'var(--on-dark-text-primary, rgba(242, 245, 247, 0.96))';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, rgba(242, 245, 247, 0.64))';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, rgba(242, 245, 247, 0.4))';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'var(--on-dark-text-paragraph, rgba(242, 245, 247, 0.8))';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = 'var(--on-dark-text-accent, #0064FE)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = 'var(--on-dark-text-negative, #FF2E43)';
/** Дополнительный акцент */
export var onDarkTextAccentDiscount = 'var(--on-dark-text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var onDarkTextAccentGold = 'var(--on-dark-text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
export var onDarkTextAccentB2B = 'var(--on-dark-text-accent-b2-b, #04C6C9)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'var(--on-light-text-primary, #24282B)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'var(--on-light-text-secondary, #8C959C)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'var(--on-light-text-tertiary, #BEC9D1)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = 'var(--on-light-text-paragraph, #24282B)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = 'var(--on-light-text-accent, #0064FE)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Дополнительный акцент */
export var onLightTextAccentDiscount = 'var(--on-light-text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var onLightTextAccentGold = 'var(--on-light-text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
export var onLightTextAccentB2B = 'var(--on-light-text-accent-b2-b, #0C8688)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = 'var(--inverse-text-primary, #24282B)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #31373B)';
/** Основной фон для карточек */
export var surfaceSolidCard = 'var(--surface-solid-card, #15181A)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = 'var(--surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = 'var(--surface-negative, #FF2E43)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentB2B = 'var(--surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
export var surfaceAccentDiscount = 'var(--surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var surfaceAccentGold = 'var(--surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
export var surfaceAccentBlue = 'var(--surface-accent-blue, #5993FF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var surfaceAdditional = 'var(--surface-additional, #494F54)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #31373B)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #15181A)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF2E43)';
/** Акцентный фон поверхности/контрола */
export var onDarkSurfaceAccentB2B = 'var(--on-dark-surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
export var onDarkSurfaceAccentDiscount = 'var(--on-dark-surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var onDarkSurfaceAccentGold = 'var(--on-dark-surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
export var onDarkSurfaceAccentBlue = 'var(--on-dark-surface-accent-blue, #5993FF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var onDarkSurfaceAdditional = 'var(--on-dark-surface-additional, #494F54)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #F2F5F7)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #E4EBF0)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = 'var(--on-light-surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = 'var(--on-light-surface-negative, #F31B31)';
/** Акцентный фон поверхности/контрола */
export var onLightSurfaceAccentB2B = 'var(--on-light-surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
export var onLightSurfaceAccentDiscount = 'var(--on-light-surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var onLightSurfaceAccentGold = 'var(--on-light-surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
export var onLightSurfaceAccentBlue = 'var(--on-light-surface-accent-blue, #3F81FD)';
/** Пунктирные дивайдеры и неактивные контролы */
export var onLightSurfaceAdditional = 'var(--on-light-surface-additional, #D1DDE5)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerRed = 'var(--banners-default-banner-red, #2E090D)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerGold = 'var(--banners-default-banner-gold, #1F1903)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerGreen = 'var(--banners-default-banner-green, #071F0C)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerMalachite = 'var(--banners-default-banner-malachite, #0A1C1C)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerBlue = 'var(--banners-default-banner-blue, #0A162B)';
/** Основной фон */
export var backgroundPrimary = 'var(--background-primary, #08090A)';
/** Вторичный фон */
export var backgroundOverlay = 'var(--background-overlay, rgba(21, 24, 26, 0.56))';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'var(--plasma-colors-white-primary, rgba(242, 245, 247, 0.96))';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'var(--plasma-colors-white-secondary, rgba(242, 245, 247, 0.64))';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'var(--plasma-colors-white-tertiary, rgba(242, 245, 247, 0.4))';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'var(--plasma-colors-black-primary, #24282B)';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'var(--plasma-colors-black-secondary, #8C959C)';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'var(--plasma-colors-black-tertiary, #BEC9D1)';
/** @deprecated instead use textPrimary */
export var text = 'var(--plasma-colors-text, #F2F5F7)';
/** @deprecated instead use textPrimary */
export var primary = 'var(--plasma-colors-primary, #F2F5F7)';
/** @deprecated instead use textSecondary */
export var secondary = 'var(--plasma-colors-secondary, #A8B4BD)';
/** @deprecated instead use textTertiary */
export var tertiary = 'var(--plasma-colors-tertiary, #7D868C)';
/** @deprecated instead use textParagraph */
export var paragraph = 'var(--plasma-colors-paragraph, #BEC9D1)';
/** @deprecated instead use backgroundPrimary */
export var background = 'var(--plasma-colors-background, #08090A)';
/** @deprecated instead use textAccent */
export var accent = 'var(--plasma-colors-accent, #0064FE)';
/** @deprecated instead use textNegative */
export var critical = 'var(--plasma-colors-critical, #FF2E43)';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02, #24282B)';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03, #31373B)';
/** @deprecated instead use textAccent */
export var buttonAccent = 'var(--plasma-colors-button-accent, #0064FE)';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = 'var(--plasma-colors-button-critical, #FF2E43)';
