// @salutejs/plasma-core/utils/formatters.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.monthShortName = exports.monthLongName = exports.monthName = exports.padZeroNumber = exports.last = void 0;

var last = function last(array) {
  return array[array.length - 1];
};

exports.last = last;

var padZeroNumber = function padZeroNumber(value) {
  return "".concat(value).padStart(2, '0');
};
/**
 * Форматирование названия месяца.
 * Если не передавать `day`, то название месяца будет, например, `июнь`.
 * В данном случае вернется `июня`.
 * Так же результат выполнения метода `formatToParts` является массивом,
 * в котором находятся части даты, форматированной на основнии `options`.
 * То есть в данном случае вернется массив [`day`, `separator`, `month`]
 */


exports.padZeroNumber = padZeroNumber;

var monthName = function monthName(val, monthFormat) {
  var targetMonth = new Date(2000, val, 1);
  return last(new Intl.DateTimeFormat('ru-RU', {
    day: 'numeric',
    month: monthFormat
  }).formatToParts(targetMonth)).value;
};

exports.monthName = monthName;

var monthLongName = function monthLongName(val) {
  return monthName(val, 'long');
};

exports.monthLongName = monthLongName;

var monthShortName = function monthShortName(val) {
  return monthName(val, 'short').replace('.', '');
};

exports.monthShortName = monthShortName;
