// @salutejs/plasma-tokens/es/brands/sberHealth/index.js
// Generated by robots, do not change this manually!
/** Цвет фона паранжи светлый на темном фоне */
export var onDarkOverlaySoft = 'var(--on-dark-overlay-soft, #0808088F)';
/** Цвет фона паранжи темный на темном фоне */
export var onDarkOverlayHard = 'var(--on-dark-overlay-hard, #080808E6)';
/** Основной цвет текста */
export var textPrimary = 'var(--text-primary, #F9F9F9FF)';
/** Вторичный цвет текста */
export var textSecondary = 'var(--text-secondary, #949494FF)';
/** Третичный цвет текста */
export var textTertiary = 'var(--text-tertiary, #4E4E4EFF)';
/** Сплошной наборный текст */
export var textParagraph = 'var(--text-paragraph, #C7C7C7FF)';
/** Акцентный цвет */
export var textAccent = 'var(--text-accent, #52BAFFFF)';
/** Акцентный цвет с градиентом */
export var textAccentGradient = 'var(--text-accent-gradient, linear-gradient(75.33deg, #4AD6FC 0%, #389AFC 100%))';
/** Цвет успеха */
export var textPositive = 'var(--text-positive, #24B23EFF)';
/** Цвет предупреждения */
export var textWarning = 'var(--text-warning, #FF7024FF)';
/** Цвет ошибки */
export var textNegative = 'var(--text-negative, #FF3D51FF)';
/** Второй акцентный цвет */
export var textAccentSecondary = 'var(--text-accent-secondary, #FF8FBCFF)';
/** Второй акцентный цвет с градиентом */
export var textAccentGradientSecondary = 'var(--text-accent-gradient-secondary, linear-gradient(75.33deg, #FFB2E5 0%, #FF4791 100%))';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'var(--on-dark-text-primary, #F9F9F9FF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, #949494FF)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, #4E4E4EFF)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'var(--on-dark-text-paragraph, #C7C7C7FF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = 'var(--on-dark-text-accent, #52BAFFFF)';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradient = 'var(--on-dark-text-accent-gradient, linear-gradient(75.33deg, #4AD6FC 0%, #389AFC 100%))';
/** Цвет успеха на темном фоне */
export var onDarkTextPositive = 'var(--on-dark-text-positive, #24B23EFF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarning = 'var(--on-dark-text-warning, #FF7024FF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = 'var(--on-dark-text-negative, #FF3D51FF)';
/** Второй акцентный цвет на темном фоне */
export var onDarkTextAccentSecondary = 'var(--on-dark-text-accent-secondary, #FF8FBCFF)';
/** Второй акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradientSecondary = 'var(--on-dark-text-accent-gradient-secondary, linear-gradient(75.33deg, #FFB2E5 0%, #FF4791 100%))';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'var(--on-light-text-primary, #171717)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'var(--on-light-text-secondary, #4E4E4E)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'var(--on-light-text-tertiary, #898989)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = 'var(--on-light-text-paragraph, #232323)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = 'var(--on-light-text-accent, #118CDFFF)';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradient = 'var(--on-light-text-accent-gradient, linear-gradient(75.33deg, #12C2F3 0%, #2281E0 100%))';
/** Цвет успеха на светлом фоне */
export var onLightTextPositive = 'var(--on-light-text-positive, #1A9E32FF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarning = 'var(--on-light-text-warning, #F55D05FF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = 'var(--on-light-text-negative, #FF293EFF)';
/** Второй акцентный цвет на светлом фоне */
export var onLightTextAccentSecondary = 'var(--on-light-text-accent-secondary, #FF4791FF)';
/** Второй акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradientSecondary = 'var(--on-light-text-accent-gradient-secondary, linear-gradient(75.33deg, #FF94C9 0%, #F00563 100%))';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = 'var(--inverse-text-primary, #171717)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondary = 'var(--inverse-text-secondary, #4E4E4E)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiary = 'var(--inverse-text-tertiary, #898989)';
/** Инвертированный cплошной наборный текст */
export var inverseTextParagraph = 'var(--inverse-text-paragraph, #232323)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности */
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
export var surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimary = 'var(--surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек */
export var surfaceTransparentCard = 'var(--surface-transparent-card, #FFFFFF0F)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = 'var(--surface-accent, #119AF0FF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%))';
/** Цвет фона поверхности/контрола успех */
export var surfacePositive = 'var(--surface-positive, #24B23EFF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarning = 'var(--surface-warning, #FF7024FF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = 'var(--surface-negative, #FF293EFF)';
/** Фон поверхности/контрола без заливки */
export var surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Второй акцентный фон поверхности/контрола */
export var surfaceAccentSecondary = 'var(--surface-accent-secondary, #FF4791FF)';
/** Второй акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientSecondary = 'var(--surface-accent-gradient-secondary, linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%))';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255, 255, 255, 0.12))';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #119AF0FF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%))';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositive = 'var(--on-dark-surface-positive, #24B23EFF)';
/** Цвет фона поверхности предупреждение на темном фоне */
export var onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FF7024FF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293EFF)';
/** Фон поверхности/контрола без заливки на темном фоне */
export var onDarkSurfaceClear = 'var(--on-dark-surface-clear, rgba(0, 0, 0, 0))';
/** Второй акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentSecondary = 'var(--on-dark-surface-accent-secondary, #FF4791FF)';
/** Второй акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradientSecondary = 'var(--on-dark-surface-accent-gradient-secondary, linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%))';
/** Непрозрачный фон поверхности по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #171717FF)';
/** Основной непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5FF)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECECFF)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDDFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFFFF)';
/** Основной прозрачный фон поверхности на светлом фоне */
export var onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, #08080808)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = 'var(--on-light-surface-accent, #119AF0FF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%))';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32FF)';
/** Цвет фона поверхности предупреждение на светлом фоне */
export var onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293EFF)';
/** Фон поверхности/контрола без заливки на светлом фоне */
export var onLightSurfaceClear = 'var(--on-light-surface-clear, rgba(0, 0, 0, 0))';
/** Второй акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentSecondary = 'var(--on-light-surface-accent-secondary, #FF4791FF)';
/** Второй акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradientSecondary = 'var(--on-light-surface-accent-gradient-secondary, linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%))';
/** Основной фон */
export var backgroundPrimary = 'var(--background-primary, #080808FF)';
/** Основной фон на темном фоне */
export var darkBackgroundPrimary = 'var(--dark-background-primary, #080808FF)';
/** Основной фон на светлом фоне */
export var lightBackgroundPrimary = 'var(--light-background-primary, #F9F9F9FF)';
/** Инвертированный основной фон */
export var inverseBackgroundPrimary = 'var(--inverse-background-primary, #F9F9F9FF)';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'var(--plasma-colors-white-primary, #F9F9F9FF)';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'var(--plasma-colors-white-secondary, #949494FF)';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'var(--plasma-colors-white-tertiary, #4E4E4EFF)';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'var(--plasma-colors-black-primary, #171717)';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'var(--plasma-colors-black-secondary, #4E4E4E)';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'var(--plasma-colors-black-tertiary, #898989)';
/** @deprecated instead use onLightSurfaceSolidDefault */
export var buttonBlack = 'var(--plasma-colors-button-black, #171717FF)';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
export var buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, rgba(8, 8, 8, 0.06))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
export var buttonWhite = 'var(--plasma-colors-button-white, #FFFFFF)';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
export var buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, rgba(255, 255, 255, 0.12))';
/** @deprecated instead use textPrimary */
export var text = 'var(--plasma-colors-text, #F9F9F9FF)';
/** @deprecated instead use textPrimary */
export var primary = 'var(--plasma-colors-primary, #F9F9F9FF)';
/** @deprecated instead use textSecondary */
export var secondary = 'var(--plasma-colors-secondary, #949494FF)';
/** @deprecated instead use textTertiary */
export var tertiary = 'var(--plasma-colors-tertiary, #4E4E4EFF)';
/** @deprecated instead use textParagraph */
export var paragraph = 'var(--plasma-colors-paragraph, #C7C7C7FF)';
/** @deprecated instead use backgroundPrimary */
export var background = 'var(--plasma-colors-background, #080808FF)';
/** @deprecated instead use textAccent */
export var accent = 'var(--plasma-colors-accent, #52BAFFFF)';
/** @deprecated instead use textPositive */
export var success = 'var(--plasma-colors-success, #24B23EFF)';
/** @deprecated instead use textWarning */
export var warning = 'var(--plasma-colors-warning, #FF7024FF)';
/** @deprecated instead use textNegative */
export var critical = 'var(--plasma-colors-critical, #FF3D51FF)';
/** @deprecated instead use surfaceTransparentPrimary */
export var surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, rgba(255, 255, 255, 0.06))';
/** @deprecated instead use surfaceTransparentSecondary */
export var surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, rgba(255, 255, 255, 0.12))';
/** @deprecated instead use surfaceTransparentTertiary */
export var surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, rgba(255, 255, 255, 0.2))';
/** @deprecated instead use surfaceSolidPrimary */
export var surfaceSolid01 = 'var(--plasma-colors-surface-solid01, #171717)';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02, #262626FF)';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03, #363636)';
/** @deprecated instead use surfaceTransparentCard */
export var surfaceCard = 'var(--plasma-colors-surface-card, #FFFFFF0F)';
/** @deprecated instead use surfaceTransparentSecondary */
export var buttonSecondary = 'var(--plasma-colors-button-secondary, rgba(255, 255, 255, 0.12))';
/** @deprecated instead use textAccent */
export var buttonAccent = 'var(--plasma-colors-button-accent, #52BAFFFF)';
/** @deprecated instead use surfacePositive */
export var buttonSuccess = 'var(--plasma-colors-button-success, #24B23EFF)';
/** @deprecated instead use surfaceWarning */
export var buttonWarning = 'var(--plasma-colors-button-warning, #FF7024FF)';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = 'var(--plasma-colors-button-critical, #FF293EFF)';
