// @salutejs/plasma-tokens/es/brands/sberHealth/values.js
// Generated by robots, do not change this manually!
/** Цвет фона паранжи светлый на темном фоне */
export var onDarkOverlaySoft = '#0808088F';
/** Цвет фона паранжи темный на темном фоне */
export var onDarkOverlayHard = '#080808E6';
/** Основной цвет текста */
export var textPrimary = '#F9F9F9FF';
/** Вторичный цвет текста */
export var textSecondary = '#949494FF';
/** Третичный цвет текста */
export var textTertiary = '#4E4E4EFF';
/** Сплошной наборный текст */
export var textParagraph = '#C7C7C7FF';
/** Акцентный цвет */
export var textAccent = '#52BAFFFF';
/** Акцентный цвет с градиентом */
export var textAccentGradient = 'linear-gradient(75.33deg, #4AD6FC 0%, #389AFC 100%)';
/** Цвет успеха */
export var textPositive = '#24B23EFF';
/** Цвет предупреждения */
export var textWarning = '#FF7024FF';
/** Цвет ошибки */
export var textNegative = '#FF3D51FF';
/** Второй акцентный цвет */
export var textAccentSecondary = '#FF8FBCFF';
/** Второй акцентный цвет с градиентом */
export var textAccentGradientSecondary = 'linear-gradient(75.33deg, #FFB2E5 0%, #FF4791 100%)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = '#F9F9F9FF';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = '#949494FF';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = '#4E4E4EFF';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = '#C7C7C7FF';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = '#52BAFFFF';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradient = 'linear-gradient(75.33deg, #4AD6FC 0%, #389AFC 100%)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositive = '#24B23EFF';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarning = '#FF7024FF';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = '#FF3D51FF';
/** Второй акцентный цвет на темном фоне */
export var onDarkTextAccentSecondary = '#FF8FBCFF';
/** Второй акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradientSecondary = 'linear-gradient(75.33deg, #FFB2E5 0%, #FF4791 100%)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = '#171717';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = '#4E4E4E';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = '#898989';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = '#232323';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = '#118CDFFF';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradient = 'linear-gradient(75.33deg, #12C2F3 0%, #2281E0 100%)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositive = '#1A9E32FF';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarning = '#F55D05FF';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = '#FF293EFF';
/** Второй акцентный цвет на светлом фоне */
export var onLightTextAccentSecondary = '#FF4791FF';
/** Второй акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradientSecondary = 'linear-gradient(75.33deg, #FF94C9 0%, #F00563 100%)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = '#171717';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondary = '#4E4E4E';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiary = '#898989';
/** Инвертированный cплошной наборный текст */
export var inverseTextParagraph = '#232323';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefault = '#FFFFFF';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimary = '#171717';
/** Вторичный непрозрачный фон поверхности */
export var surfaceSolidSecondary = '#262626FF';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = '#363636';
/** Основной фон для карточек */
export var surfaceSolidCard = '#171717';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCard = '#FFFFFF0F';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = '#119AF0FF';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositive = '#24B23EFF';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarning = '#FF7024FF';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = '#FF293EFF';
/** Фон поверхности/контрола без заливки */
export var surfaceClear = 'rgba(0, 0, 0, 0)';
/** Второй акцентный фон поверхности/контрола */
export var surfaceAccentSecondary = '#FF4791FF';
/** Второй акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientSecondary = 'linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefault = '#FFFFFF';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimary = '#171717';
/** Вторичный непрозрачный фон поверхности на темном фоне */
export var onDarkSurfaceSolidSecondary = '#262626FF';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = '#363636';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = '#171717';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCard = 'rgba(255, 255, 255, 0.12)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = '#119AF0FF';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradient = 'linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositive = '#24B23EFF';
/** Цвет фона поверхности предупреждение на темном фоне */
export var onDarkSurfaceWarning = '#FF7024FF';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = '#FF293EFF';
/** Фон поверхности/контрола без заливки на темном фоне */
export var onDarkSurfaceClear = 'rgba(0, 0, 0, 0)';
/** Второй акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentSecondary = '#FF4791FF';
/** Второй акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradientSecondary = 'linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%)';
/** Непрозрачный фон поверхности по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefault = '#171717FF';
/** Основной непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidPrimary = '#F5F5F5FF';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidSecondary = '#ECECECFF';
/** Третичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidTertiary = '#DDDDDDFF';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = '#FFFFFFFF';
/** Основной прозрачный фон поверхности на светлом фоне */
export var onLightSurfaceTransparentPrimary = '#08080808';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondary = 'rgba(8, 8, 8, 0.06)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiary = 'rgba(8, 8, 8, 0.12)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCard = '#FFFFFFFF';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = '#119AF0FF';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradient = 'linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositive = '#1A9E32FF';
/** Цвет фона поверхности предупреждение на светлом фоне */
export var onLightSurfaceWarning = '#FA5F05FF';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = '#FF293EFF';
/** Фон поверхности/контрола без заливки на светлом фоне */
export var onLightSurfaceClear = 'rgba(0, 0, 0, 0)';
/** Второй акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentSecondary = '#FF4791FF';
/** Второй акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradientSecondary = 'linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%)';
/** Основной фон */
export var backgroundPrimary = '#080808FF';
/** Основной фон на темном фоне */
export var darkBackgroundPrimary = '#080808FF';
/** Основной фон на светлом фоне */
export var lightBackgroundPrimary = '#F9F9F9FF';
/** Инвертированный основной фон */
export var inverseBackgroundPrimary = '#F9F9F9FF';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = '#F9F9F9FF';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = '#949494FF';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = '#4E4E4EFF';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = '#171717';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = '#4E4E4E';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = '#898989';
/** @deprecated instead use onLightSurfaceSolidDefault */
export var buttonBlack = '#171717FF';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
export var buttonBlackSecondary = 'rgba(8, 8, 8, 0.06)';
/** @deprecated instead use onDarkSurfaceSolidDefault */
export var buttonWhite = '#FFFFFF';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** @deprecated instead use textPrimary */
export var text = '#F9F9F9FF';
/** @deprecated instead use textPrimary */
export var primary = '#F9F9F9FF';
/** @deprecated instead use textSecondary */
export var secondary = '#949494FF';
/** @deprecated instead use textTertiary */
export var tertiary = '#4E4E4EFF';
/** @deprecated instead use textParagraph */
export var paragraph = '#C7C7C7FF';
/** @deprecated instead use backgroundPrimary */
export var background = '#080808FF';
/** @deprecated instead use textAccent */
export var accent = '#52BAFFFF';
/** @deprecated instead use textPositive */
export var success = '#24B23EFF';
/** @deprecated instead use textWarning */
export var warning = '#FF7024FF';
/** @deprecated instead use textNegative */
export var critical = '#FF3D51FF';
/** @deprecated instead use surfaceTransparentPrimary */
export var surfaceLiquid01 = 'rgba(255, 255, 255, 0.06)';
/** @deprecated instead use surfaceTransparentSecondary */
export var surfaceLiquid02 = 'rgba(255, 255, 255, 0.12)';
/** @deprecated instead use surfaceTransparentTertiary */
export var surfaceLiquid03 = 'rgba(255, 255, 255, 0.2)';
/** @deprecated instead use surfaceSolidPrimary */
export var surfaceSolid01 = '#171717';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = '#262626FF';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = '#363636';
/** @deprecated instead use surfaceTransparentCard */
export var surfaceCard = '#FFFFFF0F';
/** @deprecated instead use surfaceTransparentSecondary */
export var buttonSecondary = 'rgba(255, 255, 255, 0.12)';
/** @deprecated instead use textAccent */
export var buttonAccent = '#52BAFFFF';
/** @deprecated instead use surfacePositive */
export var buttonSuccess = '#24B23EFF';
/** @deprecated instead use surfaceWarning */
export var buttonWarning = '#FF7024FF';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = '#FF293EFF';
