// @salutejs/plasma-tokens/brands/sberprime/values.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.overlay = exports.critical = exports.warning = exports.success = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.overlayHard = exports.overlaySoft = exports.lightBackgroundPrimary = exports.darkBackgroundPrimary = exports.backgroundPrimary = exports.onLightSurfaceNegative = exports.onLightSurfaceWarning = exports.onLightSurfacePositive = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccent = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidDefault = exports.onDarkSurfaceNegative = exports.onDarkSurfaceWarning = exports.onDarkSurfacePositive = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccent = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidDefault = exports.surfaceClear = exports.surfaceNegative = exports.surfaceWarning = exports.surfacePositive = exports.surfaceAccentGradient = exports.surfaceAccent = exports.surfaceTransparentCard = exports.surfaceTransparentTertiary = exports.surfaceTransparentSecondary = exports.surfaceTransparentPrimary = exports.surfaceSolidCard = exports.surfaceSolidTertiary = exports.surfaceSolidSecondary = exports.surfaceSolidPrimary = exports.surfaceSolidDefault = exports.onLightTextNegative = exports.onLightTextWarning = exports.onLightTextPositive = exports.onLightTextAccentGradient = exports.onLightTextAccent = exports.onLightTextParagraph = exports.onLightTextTertiary = exports.onLightTextSecondary = exports.onLightTextPrimary = exports.onDarkTextNegative = exports.onDarkTextWarning = exports.onDarkTextPositive = exports.onDarkTextAccentGradient = exports.onDarkTextAccent = exports.onDarkTextParagraph = exports.onDarkTextTertiary = exports.onDarkTextSecondary = exports.onDarkTextPrimary = exports.textNegative = exports.textWarning = exports.textPositive = exports.textAccentGradient = exports.textAccent = exports.textParagraph = exports.textTertiary = exports.textSecondary = exports.textPrimary = void 0;
/** Основной цвет текста */
exports.textPrimary = 'rgba(255, 255, 255, 0.96)';
/** Вторичный цвет текста */
exports.textSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста */
exports.textTertiary = 'rgba(255, 255, 255, 0.28)';
/** Сплошной наборный текст */
exports.textParagraph = 'rgba(255, 255, 255, 0.8)';
/** Акцентный цвет */
exports.textAccent = '#15D015';
/** Акцентный цвет с градиентом */
exports.textAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет успеха */
exports.textPositive = '#1A9E32';
/** Цвет предупреждения */
exports.textWarning = '#F55D05';
/** Цвет ошибки */
exports.textNegative = '#FF293E';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'rgba(255, 255, 255, 0.96)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'rgba(255, 255, 255, 0.28)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'rgba(255, 255, 255, 0.8)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = '#15D015';
/** Акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = '#1A9E32';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = '#F55D05';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = '#FF293E';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'rgba(36, 40, 43, 0.96)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'rgba(36, 40, 43, 0.56)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'rgba(36, 40, 43, 0.28)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'rgba(36, 40, 43, 0.8)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = '#15B015';
/** Акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradient = 'radial-gradient(50% 200% at 0% 100%, rgba(250, 237, 5, 0.6) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(33, 202, 154, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(12, 156, 12, 0) 100%), #0C9C0C';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = '#1A9E32';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = '#F55D05';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = '#FF293E';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = '#F2F5F7';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = '#15181A';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = '#24282B';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = '#31373B';
/** Основной фон для карточек */
exports.surfaceSolidCard = '#15181A';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'rgba(242, 245, 247, 0.06)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'rgba(242, 245, 247, 0.12)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'rgba(242, 245, 247, 0.2)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = 'rgba(242, 245, 247, 0.06)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = '#15B015';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = '#1A9E32';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarning = '#F55D05';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = '#FF293E';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'rgba(0, 0, 0, 0)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = '#FFFFFF';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = '#15181A';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondary = '#24282B';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = '#31373B';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = '#15181A';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'rgba(255, 255, 255, 0.06)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = '#15B015';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = '#1A9E32';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceWarning = '#F55D05';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = '#FF293E';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = '#08090A';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidPrimary = '#F2F5F7';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondary = '#E4EBF0';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiary = '#D1DDE5';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = '#F2F5F7';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimary = 'rgba(8, 9, 10, 0.03)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'rgba(8, 9, 10, 0.06)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'rgba(8, 9, 10, 0.12)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = 'rgba(8, 9, 10, 0.03)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = '#15B015';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = '#1A9E32';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceWarning = '#F55D05';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = '#FF293E';
/** Основной фон */
exports.backgroundPrimary = '#08090A';
/** Основной фон на темном фоне */
exports.darkBackgroundPrimary = '#08090A';
/** Основной фон на светлом фоне */
exports.lightBackgroundPrimary = '#FFFFFF';
/** Цвет фона паранжи светлый */
exports.overlaySoft = 'rgba(8, 9, 10, 0.56)';
/** Цвет фона паранжи темный */
exports.overlayHard = 'rgba(8, 9, 10, 0.96)';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'rgba(255, 255, 255, 0.96)';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'rgba(36, 40, 43, 0.96)';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'rgba(36, 40, 43, 0.56)';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'rgba(36, 40, 43, 0.28)';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = '#08090A';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'rgba(8, 9, 10, 0.06)';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = '#FFFFFF';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** @deprecated instead use textPrimary */
exports.text = 'rgba(255, 255, 255, 0.96)';
/** @deprecated instead use textPrimary */
exports.primary = 'rgba(255, 255, 255, 0.96)';
/** @deprecated instead use textSecondary */
exports.secondary = 'rgba(255, 255, 255, 0.56)';
/** @deprecated instead use textTertiary */
exports.tertiary = 'rgba(255, 255, 255, 0.28)';
/** @deprecated instead use textParagraph */
exports.paragraph = 'rgba(255, 255, 255, 0.8)';
/** @deprecated instead use backgroundPrimary */
exports.background = '#08090A';
/** @deprecated instead use textAccent */
exports.accent = '#15D015';
/** @deprecated instead use textPositive */
exports.success = '#1A9E32';
/** @deprecated instead use textWarning */
exports.warning = '#F55D05';
/** @deprecated instead use textNegative */
exports.critical = '#FF293E';
/** @deprecated instead use overlaySoft */
exports.overlay = 'rgba(8, 9, 10, 0.56)';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'rgba(242, 245, 247, 0.06)';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'rgba(242, 245, 247, 0.12)';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'rgba(242, 245, 247, 0.2)';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = '#15181A';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = '#24282B';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = '#31373B';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = 'rgba(242, 245, 247, 0.06)';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'rgba(242, 245, 247, 0.12)';
/** @deprecated instead use textAccent */
exports.buttonAccent = '#15D015';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = '#1A9E32';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = '#F55D05';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = '#FF293E';
