// @salutejs/plasma-core/components/Basebox/Basebox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledDescription = exports.StyledLabel = exports.StyledContent = exports.StyledTrigger = exports.StyledInput = exports.StyledContentWrapper = exports.StyledRoot = void 0;

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _Typography = /*#__PURE__*/require("../Typography");

var _mixins = /*#__PURE__*/require("../../mixins");

var _tokens = /*#__PURE__*/require("../../tokens");

var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

(0, _collectPackageInfo.collectPackageInfo)('Basebox/Basebox');

var StyledRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-11r48g6-0"
})(["", " position:relative;align-items:flex-start;display:flex;cursor:pointer;"], _mixins.applyNoSelect);

exports.StyledRoot = StyledRoot;

var StyledContentWrapper = /*#__PURE__*/_styledComponents["default"].label.withConfig({
  componentId: "plasma__sc-11r48g6-1"
})(["position:relative;align-items:flex-start;display:flex;width:100%;cursor:inherit;"]);

exports.StyledContentWrapper = StyledContentWrapper;

var StyledInput = /*#__PURE__*/_styledComponents["default"].input.withConfig({
  componentId: "plasma__sc-11r48g6-2"
})(["position:absolute;&:focus{outline:0 none;}"]);

exports.StyledInput = StyledInput;

var StyledTrigger = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-11r48g6-3"
})(["box-sizing:border-box;position:relative;transition:", ";"], function (_ref) {
  var theme = _ref.theme;
  return theme.lowPerformance ? 'unset' : 'all 0.1s ease-in-out';
});

exports.StyledTrigger = StyledTrigger;

var StyledContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-11r48g6-4"
})(["display:flex;flex-direction:column;margin-left:0.75rem;"]);

exports.StyledContent = StyledContent;
var StyledLabel = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body1).withConfig({
  componentId: "plasma__sc-11r48g6-5"
})(["", ""], _mixins.applyEllipsis);
exports.StyledLabel = StyledLabel;
var StyledDescription = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote1).withConfig({
  componentId: "plasma__sc-11r48g6-6"
})(["color:", ";"], _tokens.secondary);
exports.StyledDescription = StyledDescription;
