// @salutejs/plasma-tokens/es/brands/sberprime/values.js
// Generated by robots, do not change this manually!
/** Основной цвет текста */
export var textPrimary = 'rgba(255, 255, 255, 0.96)';
/** Вторичный цвет текста */
export var textSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста */
export var textTertiary = 'rgba(255, 255, 255, 0.28)';
/** Сплошной наборный текст */
export var textParagraph = 'rgba(255, 255, 255, 0.8)';
/** Акцентный цвет */
export var textAccent = '#15D015';
/** Акцентный цвет с градиентом */
export var textAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет успеха */
export var textPositive = '#1A9E32';
/** Цвет предупреждения */
export var textWarning = '#F55D05';
/** Цвет ошибки */
export var textNegative = '#FF293E';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'rgba(255, 255, 255, 0.96)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'rgba(255, 255, 255, 0.28)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'rgba(255, 255, 255, 0.8)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = '#15D015';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет успеха на темном фоне */
export var onDarkTextPositive = '#1A9E32';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarning = '#F55D05';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = '#FF293E';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'rgba(36, 40, 43, 0.96)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'rgba(36, 40, 43, 0.56)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'rgba(36, 40, 43, 0.28)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = 'rgba(36, 40, 43, 0.8)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = '#15B015';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradient = 'radial-gradient(50% 200% at 0% 100%, rgba(250, 237, 5, 0.6) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(33, 202, 154, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(12, 156, 12, 0) 100%), #0C9C0C';
/** Цвет успеха на светлом фоне */
export var onLightTextPositive = '#1A9E32';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarning = '#F55D05';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = '#FF293E';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefault = '#F2F5F7';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimary = '#15181A';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = '#24282B';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = '#31373B';
/** Основной фон для карточек */
export var surfaceSolidCard = '#15181A';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimary = 'rgba(242, 245, 247, 0.06)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondary = 'rgba(242, 245, 247, 0.12)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiary = 'rgba(242, 245, 247, 0.2)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCard = 'rgba(242, 245, 247, 0.06)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = '#15B015';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет фона поверхности/контрола успех */
export var surfacePositive = '#1A9E32';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarning = '#F55D05';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = '#FF293E';
/** Фон поверхности/контрола без заливки */
export var surfaceClear = 'rgba(0, 0, 0, 0)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefault = '#FFFFFF';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimary = '#15181A';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondary = '#24282B';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = '#31373B';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = '#15181A';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCard = 'rgba(255, 255, 255, 0.06)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = '#15B015';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositive = '#1A9E32';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarning = '#F55D05';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = '#FF293E';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefault = '#08090A';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimary = '#F2F5F7';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondary = '#E4EBF0';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiary = '#D1DDE5';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = '#F2F5F7';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimary = 'rgba(8, 9, 10, 0.03)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondary = 'rgba(8, 9, 10, 0.06)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiary = 'rgba(8, 9, 10, 0.12)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCard = 'rgba(8, 9, 10, 0.03)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = '#15B015';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradient = 'radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositive = '#1A9E32';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarning = '#F55D05';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = '#FF293E';
/** Основной фон */
export var backgroundPrimary = '#08090A';
/** Основной фон на темном фоне */
export var darkBackgroundPrimary = '#08090A';
/** Основной фон на светлом фоне */
export var lightBackgroundPrimary = '#FFFFFF';
/** Цвет фона паранжи светлый */
export var overlaySoft = 'rgba(8, 9, 10, 0.56)';
/** Цвет фона паранжи темный */
export var overlayHard = 'rgba(8, 9, 10, 0.96)';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'rgba(255, 255, 255, 0.96)';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'rgba(36, 40, 43, 0.96)';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'rgba(36, 40, 43, 0.56)';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'rgba(36, 40, 43, 0.28)';
/** @deprecated instead use onLightSurfaceSolidDefault */
export var buttonBlack = '#08090A';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
export var buttonBlackSecondary = 'rgba(8, 9, 10, 0.06)';
/** @deprecated instead use onDarkSurfaceSolidDefault */
export var buttonWhite = '#FFFFFF';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** @deprecated instead use textPrimary */
export var text = 'rgba(255, 255, 255, 0.96)';
/** @deprecated instead use textPrimary */
export var primary = 'rgba(255, 255, 255, 0.96)';
/** @deprecated instead use textSecondary */
export var secondary = 'rgba(255, 255, 255, 0.56)';
/** @deprecated instead use textTertiary */
export var tertiary = 'rgba(255, 255, 255, 0.28)';
/** @deprecated instead use textParagraph */
export var paragraph = 'rgba(255, 255, 255, 0.8)';
/** @deprecated instead use backgroundPrimary */
export var background = '#08090A';
/** @deprecated instead use textAccent */
export var accent = '#15D015';
/** @deprecated instead use textPositive */
export var success = '#1A9E32';
/** @deprecated instead use textWarning */
export var warning = '#F55D05';
/** @deprecated instead use textNegative */
export var critical = '#FF293E';
/** @deprecated instead use overlaySoft */
export var overlay = 'rgba(8, 9, 10, 0.56)';
/** @deprecated instead use surfaceTransparentPrimary */
export var surfaceLiquid01 = 'rgba(242, 245, 247, 0.06)';
/** @deprecated instead use surfaceTransparentSecondary */
export var surfaceLiquid02 = 'rgba(242, 245, 247, 0.12)';
/** @deprecated instead use surfaceTransparentTertiary */
export var surfaceLiquid03 = 'rgba(242, 245, 247, 0.2)';
/** @deprecated instead use surfaceSolidPrimary */
export var surfaceSolid01 = '#15181A';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = '#24282B';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = '#31373B';
/** @deprecated instead use surfaceTransparentCard */
export var surfaceCard = 'rgba(242, 245, 247, 0.06)';
/** @deprecated instead use surfaceTransparentSecondary */
export var buttonSecondary = 'rgba(242, 245, 247, 0.12)';
/** @deprecated instead use textAccent */
export var buttonAccent = '#15D015';
/** @deprecated instead use surfacePositive */
export var buttonSuccess = '#1A9E32';
/** @deprecated instead use surfaceWarning */
export var buttonWarning = '#F55D05';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = '#FF293E';
