// @salutejs/plasma-ui/components/ProductCard/ProductCardPrice.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ProductCardPrice = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");

var _Price = /*#__PURE__*/require("../Price");

var _Typography = /*#__PURE__*/require("../Typography");

var _types = /*#__PURE__*/require("./types");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var StyledPrices = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-eqx89j-0"
})(["display:flex;flex-direction:column;justify-content:center;width:100%;min-height:2.375rem;"]);

var StyledPrice = /*#__PURE__*/(0, _styledComponents["default"])(_Price.Price).withConfig({
  componentId: "plasma-ui__sc-eqx89j-1"
})(["color:", ";", ""], function (_ref) {
  var $type = _ref.$type;

  switch ($type) {
    case _types.PriceTypes.NEW:
      return _plasmaCore.success;

    default:
      return _plasmaCore.black;
  }
}, function (_ref2) {
  var $type = _ref2.$type;
  return $type === _types.PriceTypes.OLD && (0, _styledComponents.css)(["opacity:0.56;"]);
});
var ProductCardPrice = /*#__PURE__*/(0, _react.memo)(function (_ref3) {
  var price = _ref3.price,
      oldPrice = _ref3.oldPrice,
      periodicity = _ref3.periodicity;

  if (!price) {
    return null;
  }

  return /*#__PURE__*/_react["default"].createElement(StyledPrices, null, /*#__PURE__*/_react["default"].createElement(StyledPrice, {
    $type: oldPrice !== undefined ? _types.PriceTypes.NEW : undefined,
    forwardedAs: _Typography.Body2,
    periodicity: periodicity
  }, price), oldPrice && /*#__PURE__*/_react["default"].createElement(StyledPrice, {
    $type: _types.PriceTypes.OLD,
    stroke: true,
    forwardedAs: _Typography.Caption,
    periodicity: periodicity
  }, oldPrice));
});
exports.ProductCardPrice = ProductCardPrice;
