// @salutejs/plasma-core/tokens/colors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.skeletonGradientLighter = exports.skeletonGradient = exports.voicePhraseGradient = exports.gradient = exports.buttonFocused = exports.buttonAccent = exports.accent = exports.speechBubbleReceived = exports.speechBubbleSent = exports.buttonChecked = exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonSecondary = exports.buttonPrimary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.overlay = exports.critical = exports.warning = exports.success = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.background = exports.paragraph = exports.inverse = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.transparent = exports.dark03 = exports.dark02 = exports.dark01 = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.black = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.white = void 0;
var white = 'var(--plasma-colors-white)';
exports.white = white;
var whitePrimary = 'var(--plasma-colors-white-primary)';
exports.whitePrimary = whitePrimary;
var whiteSecondary = 'var(--plasma-colors-white-secondary)';
exports.whiteSecondary = whiteSecondary;
var whiteTertiary = 'var(--plasma-colors-white-tertiary)';
exports.whiteTertiary = whiteTertiary;
var black = 'var(--plasma-colors-black)';
exports.black = black;
var blackPrimary = 'var(--plasma-colors-black-primary)';
exports.blackPrimary = blackPrimary;
var blackSecondary = 'var(--plasma-colors-black-secondary)';
exports.blackSecondary = blackSecondary;
var blackTertiary = 'var(--plasma-colors-black-tertiary)';
exports.blackTertiary = blackTertiary;
var dark01 = 'var(--plasma-colors-dark01)';
exports.dark01 = dark01;
var dark02 = 'var(--plasma-colors-dark02)';
exports.dark02 = dark02;
var dark03 = 'var(--plasma-colors-dark03)';
exports.dark03 = dark03;
var transparent = 'var(--plasma-colors-transparent)';
exports.transparent = transparent;
var text = 'var(--plasma-colors-text)';
exports.text = text;
var primary = 'var(--plasma-colors-primary)';
exports.primary = primary;
var secondary = 'var(--plasma-colors-secondary)';
exports.secondary = secondary;
var tertiary = 'var(--plasma-colors-tertiary)';
exports.tertiary = tertiary;
var inverse = 'var(--plasma-colors-inverse)';
exports.inverse = inverse;
var paragraph = 'var(--plasma-colors-paragraph)';
exports.paragraph = paragraph;
var background = 'var(--plasma-colors-background)';
exports.background = background;
var backgroundPrimary = 'var(--plasma-colors-background-primary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107

exports.backgroundPrimary = backgroundPrimary;
var backgroundSecondary = 'var(--plasma-colors-background-secondary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107

exports.backgroundSecondary = backgroundSecondary;
var backgroundTertiary = 'var(--plasma-colors-background-tertiary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107

exports.backgroundTertiary = backgroundTertiary;
var success = 'var(--plasma-colors-success)';
exports.success = success;
var warning = 'var(--plasma-colors-warning)';
exports.warning = warning;
var critical = 'var(--plasma-colors-critical)';
exports.critical = critical;
var overlay = 'var(--plasma-colors-overlay)';
exports.overlay = overlay;
var surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107

exports.surfaceLiquid01 = surfaceLiquid01;
var surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107

exports.surfaceLiquid02 = surfaceLiquid02;
var surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107

exports.surfaceLiquid03 = surfaceLiquid03;
var surfaceSolid01 = 'var(--plasma-colors-surface-solid01)';
exports.surfaceSolid01 = surfaceSolid01;
var surfaceSolid02 = 'var(--plasma-colors-surface-solid02)';
exports.surfaceSolid02 = surfaceSolid02;
var surfaceSolid03 = 'var(--plasma-colors-surface-solid03)';
exports.surfaceSolid03 = surfaceSolid03;
var surfaceCard = 'var(--plasma-colors-surface-card)';
exports.surfaceCard = surfaceCard;
var buttonPrimary = 'var(--plasma-colors-button-primary)';
exports.buttonPrimary = buttonPrimary;
var buttonSecondary = 'var(--plasma-colors-button-secondary)';
exports.buttonSecondary = buttonSecondary;
var buttonSuccess = 'var(--plasma-colors-button-success)';
exports.buttonSuccess = buttonSuccess;
var buttonWarning = 'var(--plasma-colors-button-warning)';
exports.buttonWarning = buttonWarning;
var buttonCritical = 'var(--plasma-colors-button-critical)';
exports.buttonCritical = buttonCritical;
var buttonChecked = 'var(--plasma-colors-button-checked)';
exports.buttonChecked = buttonChecked;
var speechBubbleSent = 'var(--plasma-colors-speech-bubble-sent)';
exports.speechBubbleSent = speechBubbleSent;
var speechBubbleReceived = 'var(--plasma-colors-speech-bubble-received)';
exports.speechBubbleReceived = speechBubbleReceived;
var accent = 'var(--plasma-colors-accent)';
exports.accent = accent;
var buttonAccent = 'var(--plasma-colors-button-accent)';
exports.buttonAccent = buttonAccent;
var buttonFocused = 'var(--plasma-colors-button-focused)';
exports.buttonFocused = buttonFocused;
var gradient = 'var(--plasma-colors-gradient)';
exports.gradient = gradient;
var voicePhraseGradient = 'var(--plasma-colors-voice-phrase-gradient)';
exports.voicePhraseGradient = voicePhraseGradient;
var skeletonGradient = 'var(--plasma-colors-skeleton-gradient)';
exports.skeletonGradient = skeletonGradient;
var skeletonGradientLighter = 'var(--plasma-colors-skeleton-gradient-lighter)';
exports.skeletonGradientLighter = skeletonGradientLighter;
