// @salutejs/plasma-tokens/es/themesValues/salutejs_brand__dark.js
// Generated by robots, do not change this manually!
/** Базовый белый цвет совпадает с whitePrimary */
export var white = '#FFFFFF';
/** Первичный белый цвет */
export var whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
export var whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
export var whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
export var black = '#080808';
/** Первичный черный цвет */
export var blackPrimary = '#080808';
/** Вторичный черныйцвет */
export var blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
export var blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
export var dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
export var dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
export var dark03 = '#363636';
/** Прозрачный цвет */
export var transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
export var buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
export var buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
export var buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
export var buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
export var buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
export var text = 'rgba(255, 255, 255, 0.96)';
/** Основной текст в интерфейсе */
export var primary = 'rgba(255, 255, 255, 0.96)';
/** Второстепенный текст */
export var secondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста */
export var tertiary = 'rgba(255, 255, 255, 0.28)';
/** Сплошной наборный текст */
export var paragraph = 'rgba(255, 255, 255, 0.96)';
/** Белый в светлой теме / Чёрный в тёмной теме */
export var inverse = '#080808';
/** Основной бэкграунд всех приложений и смартапов */
export var background = '#080808';
/** Первичный цвет фона */
export var backgroundPrimary = '#171717';
/** Вторичный цвет фона */
export var backgroundSecondary = '#232323';
/** Третичный цвет фона */
export var backgroundTertiary = '#363636';
/** Обозначение успешного сценария */
export var success = '#24B23E';
/** Цвет предупреждения */
export var warning = '#FA6D20';
/** Цвет ошибки */
export var critical = '#FF4D5F';
/** Полупрозрачные поверхности */
export var surfaceLiquid01 = 'rgba(255, 255, 255, 0.06)';
/** Полупрозрачные поверхности */
export var surfaceLiquid02 = 'rgba(255, 255, 255, 0.12)';
/** Полупрозрачные поверхности */
export var surfaceLiquid03 = 'rgba(255, 255, 255, 0.2)';
/** Непрозрачные поверхности */
export var surfaceSolid01 = '#171717';
/** Непрозрачные поверхности */
export var surfaceSolid02 = '#232323';
/** Непрозрачные поверхности */
export var surfaceSolid03 = '#363636';
/** Основной фон для карточек */
export var surfaceCard = 'rgba(255, 255, 255, 0.12)';
/** Первичный цвет контролов */
export var buttonPrimary = '#FFFFFF';
/** Второстепенная кнопка */
export var buttonSecondary = 'rgba(255, 255, 255, 0.12)';
/** Кнопка для успешного сценария */
export var buttonSuccess = '#21A038';
/** Цвет предупреждения у контролов */
export var buttonWarning = '#E35502';
/** Цвет ошибки у контролов */
export var buttonCritical = '#FF2E43';
/** Цвет зажатого контрола */
export var buttonChecked = '#FFFFFF';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
/** Цвет фона баблов отправленный сообщений */
export var speechBubbleSent = 'rgba(0, 0, 0, 0.28)';
/** Цвет фона баблов получнных сообщений */
export var speechBubbleReceived = 'rgba(255, 255, 255, 0.12)';
/** Цвет фона паранжи */
export var overlay = 'rgba(0, 0, 0, 0.8)';
export var overlaySoft = 'rgba(8, 8, 8, 0.56)';
export var overlayHard = 'rgba(8, 8, 8, 0.9)';
export var overlayBlur = 'rgba(23, 23, 23, 0.28)';
export var darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var darkOverlayBlur = 'rgba(23, 23, 23, 0.28)';
export var lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var lightOverlayBlur = 'rgba(23, 23, 23, 0.28)';
export var inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var inverseOverlayBlur = 'rgba(23, 23, 23, 0.28)';
/** Акцентный цвет бренда */
export var accent = '#5993FF';
/** Акцентный цвет у контролов */
export var buttonAccent = '#3F82FD';
/** Цвет рамки фокуса у контрола */
export var buttonFocused = 'rgba(255, 255, 255, 0.96)';
/** Градиент для заливки основного фона */
export var gradient = '#080808';
/** Градиент для заливки основного фона */
export var gradientDevice = '#080808';
/** Градиент подсказок о голосовых запросах */
export var voicePhraseGradient = 'linear-gradient(45deg, hsla(219.87577639751552, 100%, 68.43137254901961%, 1) 0%, hsla(190.1639344262295, 94.81865284974093%, 62.15686274509804%, 1) 100%)';
