// @salutejs/plasma-core/components/Badge/Badge.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Badge = exports.badgeRootSizes = exports.badgeSizes = void 0;

var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _tokens = /*#__PURE__*/require("../../tokens");

var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");

var _excluded = ["size", "text", "contentLeft"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

(0, _collectPackageInfo.collectPackageInfo)('Badge/Badge');

/**
 * Размеры в ремах.
 */
var badgeSizes = {
  l: {
    textMarginX: '0.25rem',
    textMarginLeftAfterContent: '0.25rem'
  },
  s: {
    textMarginX: '0.125rem',
    textMarginLeftAfterContent: '0.125rem'
  }
};
exports.badgeSizes = badgeSizes;
var badgeRootSizes = {
  l: {
    height: '1.5rem',
    padding: '0.25rem',
    minWidth: '1.5rem',
    borderRadius: '0.75rem',
    fontSize: '0.75rem'
  },
  s: {
    height: '1rem',
    padding: '0.125rem',
    minWidth: '1rem',
    borderRadius: '0.5rem',
    fontSize: '0.625rem'
  }
};
exports.badgeRootSizes = badgeRootSizes;

var StyledText = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma__sc-16llsfu-0"
})([""]);

var StyledContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-16llsfu-1"
})(["display:flex;"]);

var StyledBadge = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-16llsfu-2"
})(["", ";display:inline-flex;align-items:center;box-sizing:border-box;justify-content:center;width:max-content;", ";"], _tokens.caption, function (_ref) {
  var size = _ref.size,
      circled = _ref.circled;
  return (0, _styledComponents.css)(["", ";", " ", "{margin-left:", ";margin-right:", ";}", " ~ ", "{margin-left:", ";}"], badgeRootSizes[size], circled && (0, _styledComponents.css)(["width:", ";padding-left:0;padding-right:0;"], badgeRootSizes[size].minWidth), StyledText, badgeSizes[size].textMarginX, badgeSizes[size].textMarginX, StyledContent, StyledText, badgeSizes[size].textMarginLeftAfterContent);
});

/**
 * Небольшая бирка для ячеек и карточек.
 * Компонент может отображаться в нескольких размерах и цветах, может содержать текст и/или иконку.
 */
var Badge = function Badge(_ref2) {
  var _ref2$size = _ref2.size,
      size = _ref2$size === void 0 ? 'l' : _ref2$size,
      text = _ref2.text,
      contentLeft = _ref2.contentLeft,
      rest = _objectWithoutProperties(_ref2, _excluded);

  return /*#__PURE__*/_react["default"].createElement(StyledBadge, _extends({
    size: size
  }, rest), contentLeft && /*#__PURE__*/_react["default"].createElement(StyledContent, null, contentLeft), text && /*#__PURE__*/_react["default"].createElement(StyledText, null, text));
};

exports.Badge = Badge;
