// @salutejs/plasma-tokens/brands/sberHealth/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.inverseBackgroundPrimary = exports.lightBackgroundPrimary = exports.darkBackgroundPrimary = exports.backgroundPrimary = exports.onLightSurfaceAccentGradientSecondary = exports.onLightSurfaceAccentSecondary = exports.onLightSurfaceClear = exports.onLightSurfaceNegative = exports.onLightSurfaceWarning = exports.onLightSurfacePositive = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccent = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidDefault = exports.onDarkSurfaceAccentGradientSecondary = exports.onDarkSurfaceAccentSecondary = exports.onDarkSurfaceClear = exports.onDarkSurfaceNegative = exports.onDarkSurfaceWarning = exports.onDarkSurfacePositive = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccent = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidDefault = exports.surfaceAccentGradientSecondary = exports.surfaceAccentSecondary = exports.surfaceClear = exports.surfaceNegative = exports.surfaceWarning = exports.surfacePositive = exports.surfaceAccentGradient = exports.surfaceAccent = exports.surfaceTransparentCard = exports.surfaceTransparentTertiary = exports.surfaceTransparentSecondary = exports.surfaceTransparentPrimary = exports.surfaceSolidCard = exports.surfaceSolidTertiary = exports.surfaceSolidSecondary = exports.surfaceSolidPrimary = exports.surfaceSolidDefault = exports.inverseTextParagraph = exports.inverseTextTertiary = exports.inverseTextSecondary = exports.inverseTextPrimary = exports.onLightTextAccentGradientSecondary = exports.onLightTextAccentSecondary = exports.onLightTextNegative = exports.onLightTextWarning = exports.onLightTextPositive = exports.onLightTextAccentGradient = exports.onLightTextAccent = exports.onLightTextParagraph = exports.onLightTextTertiary = exports.onLightTextSecondary = exports.onLightTextPrimary = exports.onDarkTextAccentGradientSecondary = exports.onDarkTextAccentSecondary = exports.onDarkTextNegative = exports.onDarkTextWarning = exports.onDarkTextPositive = exports.onDarkTextAccentGradient = exports.onDarkTextAccent = exports.onDarkTextParagraph = exports.onDarkTextTertiary = exports.onDarkTextSecondary = exports.onDarkTextPrimary = exports.textAccentGradientSecondary = exports.textAccentSecondary = exports.textNegative = exports.textWarning = exports.textPositive = exports.textAccentGradient = exports.textAccent = exports.textParagraph = exports.textTertiary = exports.textSecondary = exports.textPrimary = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = void 0;
/** Цвет фона паранжи светлый на темном фоне */
exports.onDarkOverlaySoft = 'var(--on-dark-overlay-soft, #0808088F)';
/** Цвет фона паранжи темный на темном фоне */
exports.onDarkOverlayHard = 'var(--on-dark-overlay-hard, #080808E6)';
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #F9F9F9FF)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #949494FF)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #4E4E4EFF)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, #C7C7C7FF)';
/** Акцентный цвет */
exports.textAccent = 'var(--text-accent, #52BAFFFF)';
/** Акцентный цвет с градиентом */
exports.textAccentGradient = 'var(--text-accent-gradient, linear-gradient(75.33deg, #4AD6FC 0%, #389AFC 100%))';
/** Цвет успеха */
exports.textPositive = 'var(--text-positive, #24B23EFF)';
/** Цвет предупреждения */
exports.textWarning = 'var(--text-warning, #FF7024FF)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF3D51FF)';
/** Второй акцентный цвет */
exports.textAccentSecondary = 'var(--text-accent-secondary, #FF8FBCFF)';
/** Второй акцентный цвет с градиентом */
exports.textAccentGradientSecondary = 'var(--text-accent-gradient-secondary, linear-gradient(75.33deg, #FFB2E5 0%, #FF4791 100%))';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, #F9F9F9FF)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, #949494FF)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, #4E4E4EFF)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, #C7C7C7FF)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = 'var(--on-dark-text-accent, #52BAFFFF)';
/** Акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradient = 'var(--on-dark-text-accent-gradient, linear-gradient(75.33deg, #4AD6FC 0%, #389AFC 100%))';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = 'var(--on-dark-text-positive, #24B23EFF)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = 'var(--on-dark-text-warning, #FF7024FF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF3D51FF)';
/** Второй акцентный цвет на темном фоне */
exports.onDarkTextAccentSecondary = 'var(--on-dark-text-accent-secondary, #FF8FBCFF)';
/** Второй акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradientSecondary = 'var(--on-dark-text-accent-gradient-secondary, linear-gradient(75.33deg, #FFB2E5 0%, #FF4791 100%))';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, #171717)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, #4E4E4E)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #898989)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, #232323)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = 'var(--on-light-text-accent, #118CDFFF)';
/** Акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradient = 'var(--on-light-text-accent-gradient, linear-gradient(75.33deg, #12C2F3 0%, #2281E0 100%))';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = 'var(--on-light-text-positive, #1A9E32FF)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = 'var(--on-light-text-warning, #F55D05FF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #FF293EFF)';
/** Второй акцентный цвет на светлом фоне */
exports.onLightTextAccentSecondary = 'var(--on-light-text-accent-secondary, #FF4791FF)';
/** Второй акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradientSecondary = 'var(--on-light-text-accent-gradient-secondary, linear-gradient(75.33deg, #FF94C9 0%, #F00563 100%))';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, #171717)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = 'var(--inverse-text-secondary, #4E4E4E)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = 'var(--inverse-text-tertiary, #898989)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraph = 'var(--inverse-text-paragraph, #232323)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'var(--surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = 'var(--surface-transparent-card, #FFFFFF0F)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = 'var(--surface-accent, #119AF0FF)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%))';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = 'var(--surface-positive, #24B23EFF)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarning = 'var(--surface-warning, #FF7024FF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF293EFF)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Второй акцентный фон поверхности/контрола */
exports.surfaceAccentSecondary = 'var(--surface-accent-secondary, #FF4791FF)';
/** Второй акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradientSecondary = 'var(--surface-accent-gradient-secondary, linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%))';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255, 255, 255, 0.12))';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #119AF0FF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%))';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = 'var(--on-dark-surface-positive, #24B23EFF)';
/** Цвет фона поверхности предупреждение на темном фоне */
exports.onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FF7024FF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293EFF)';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClear = 'var(--on-dark-surface-clear, rgba(0, 0, 0, 0))';
/** Второй акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentSecondary = 'var(--on-dark-surface-accent-secondary, #FF4791FF)';
/** Второй акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradientSecondary = 'var(--on-dark-surface-accent-gradient-secondary, linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%))';
/** Непрозрачный фон поверхности по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #171717FF)';
/** Основной непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5FF)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECECFF)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDDFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFFFF)';
/** Основной прозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, #08080808)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = 'var(--on-light-surface-accent, #119AF0FF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%))';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32FF)';
/** Цвет фона поверхности предупреждение на светлом фоне */
exports.onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293EFF)';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClear = 'var(--on-light-surface-clear, rgba(0, 0, 0, 0))';
/** Второй акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentSecondary = 'var(--on-light-surface-accent-secondary, #FF4791FF)';
/** Второй акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradientSecondary = 'var(--on-light-surface-accent-gradient-secondary, linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%))';
/** Основной фон */
exports.backgroundPrimary = 'var(--background-primary, #080808FF)';
/** Основной фон на темном фоне */
exports.darkBackgroundPrimary = 'var(--dark-background-primary, #080808FF)';
/** Основной фон на светлом фоне */
exports.lightBackgroundPrimary = 'var(--light-background-primary, #F9F9F9FF)';
/** Инвертированный основной фон */
exports.inverseBackgroundPrimary = 'var(--inverse-background-primary, #F9F9F9FF)';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, #F9F9F9FF)';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, #949494FF)';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, #4E4E4EFF)';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, #171717)';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, #4E4E4E)';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, #898989)';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = 'var(--plasma-colors-button-black, #171717FF)';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, rgba(8, 8, 8, 0.06))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = 'var(--plasma-colors-button-white, #FFFFFF)';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, rgba(255, 255, 255, 0.12))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, #F9F9F9FF)';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, #F9F9F9FF)';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, #949494FF)';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, #4E4E4EFF)';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, #C7C7C7FF)';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, #080808FF)';
/** @deprecated instead use textAccent */
exports.accent = 'var(--plasma-colors-accent, #52BAFFFF)';
/** @deprecated instead use textPositive */
exports.success = 'var(--plasma-colors-success, #24B23EFF)';
/** @deprecated instead use textWarning */
exports.warning = 'var(--plasma-colors-warning, #FF7024FF)';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, #FF3D51FF)';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, rgba(255, 255, 255, 0.06))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, rgba(255, 255, 255, 0.12))';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, rgba(255, 255, 255, 0.2))';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, #171717)';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, #262626FF)';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, #363636)';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = 'var(--plasma-colors-surface-card, #FFFFFF0F)';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'var(--plasma-colors-button-secondary, rgba(255, 255, 255, 0.12))';
/** @deprecated instead use textAccent */
exports.buttonAccent = 'var(--plasma-colors-button-accent, #52BAFFFF)';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = 'var(--plasma-colors-button-success, #24B23EFF)';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = 'var(--plasma-colors-button-warning, #FF7024FF)';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, #FF293EFF)';
