// @salutejs/plasma-tokens/brands/sberHealth/values.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.inverseBackgroundPrimary = exports.lightBackgroundPrimary = exports.darkBackgroundPrimary = exports.backgroundPrimary = exports.onLightSurfaceAccentGradientSecondary = exports.onLightSurfaceAccentSecondary = exports.onLightSurfaceClear = exports.onLightSurfaceNegative = exports.onLightSurfaceWarning = exports.onLightSurfacePositive = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccent = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidDefault = exports.onDarkSurfaceAccentGradientSecondary = exports.onDarkSurfaceAccentSecondary = exports.onDarkSurfaceClear = exports.onDarkSurfaceNegative = exports.onDarkSurfaceWarning = exports.onDarkSurfacePositive = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccent = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidDefault = exports.surfaceAccentGradientSecondary = exports.surfaceAccentSecondary = exports.surfaceClear = exports.surfaceNegative = exports.surfaceWarning = exports.surfacePositive = exports.surfaceAccentGradient = exports.surfaceAccent = exports.surfaceTransparentCard = exports.surfaceTransparentTertiary = exports.surfaceTransparentSecondary = exports.surfaceTransparentPrimary = exports.surfaceSolidCard = exports.surfaceSolidTertiary = exports.surfaceSolidSecondary = exports.surfaceSolidPrimary = exports.surfaceSolidDefault = exports.inverseTextParagraph = exports.inverseTextTertiary = exports.inverseTextSecondary = exports.inverseTextPrimary = exports.onLightTextAccentGradientSecondary = exports.onLightTextAccentSecondary = exports.onLightTextNegative = exports.onLightTextWarning = exports.onLightTextPositive = exports.onLightTextAccentGradient = exports.onLightTextAccent = exports.onLightTextParagraph = exports.onLightTextTertiary = exports.onLightTextSecondary = exports.onLightTextPrimary = exports.onDarkTextAccentGradientSecondary = exports.onDarkTextAccentSecondary = exports.onDarkTextNegative = exports.onDarkTextWarning = exports.onDarkTextPositive = exports.onDarkTextAccentGradient = exports.onDarkTextAccent = exports.onDarkTextParagraph = exports.onDarkTextTertiary = exports.onDarkTextSecondary = exports.onDarkTextPrimary = exports.textAccentGradientSecondary = exports.textAccentSecondary = exports.textNegative = exports.textWarning = exports.textPositive = exports.textAccentGradient = exports.textAccent = exports.textParagraph = exports.textTertiary = exports.textSecondary = exports.textPrimary = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = void 0;
/** Цвет фона паранжи светлый на темном фоне */
exports.onDarkOverlaySoft = '#0808088F';
/** Цвет фона паранжи темный на темном фоне */
exports.onDarkOverlayHard = '#080808E6';
/** Основной цвет текста */
exports.textPrimary = '#F9F9F9FF';
/** Вторичный цвет текста */
exports.textSecondary = '#949494FF';
/** Третичный цвет текста */
exports.textTertiary = '#4E4E4EFF';
/** Сплошной наборный текст */
exports.textParagraph = '#C7C7C7FF';
/** Акцентный цвет */
exports.textAccent = '#52BAFFFF';
/** Акцентный цвет с градиентом */
exports.textAccentGradient = 'linear-gradient(75.33deg, #4AD6FC 0%, #389AFC 100%)';
/** Цвет успеха */
exports.textPositive = '#24B23EFF';
/** Цвет предупреждения */
exports.textWarning = '#FF7024FF';
/** Цвет ошибки */
exports.textNegative = '#FF3D51FF';
/** Второй акцентный цвет */
exports.textAccentSecondary = '#FF8FBCFF';
/** Второй акцентный цвет с градиентом */
exports.textAccentGradientSecondary = 'linear-gradient(75.33deg, #FFB2E5 0%, #FF4791 100%)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = '#F9F9F9FF';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = '#949494FF';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = '#4E4E4EFF';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = '#C7C7C7FF';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = '#52BAFFFF';
/** Акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradient = 'linear-gradient(75.33deg, #4AD6FC 0%, #389AFC 100%)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = '#24B23EFF';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = '#FF7024FF';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = '#FF3D51FF';
/** Второй акцентный цвет на темном фоне */
exports.onDarkTextAccentSecondary = '#FF8FBCFF';
/** Второй акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradientSecondary = 'linear-gradient(75.33deg, #FFB2E5 0%, #FF4791 100%)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = '#171717';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = '#4E4E4E';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = '#898989';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = '#232323';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = '#118CDFFF';
/** Акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradient = 'linear-gradient(75.33deg, #12C2F3 0%, #2281E0 100%)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = '#1A9E32FF';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = '#F55D05FF';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = '#FF293EFF';
/** Второй акцентный цвет на светлом фоне */
exports.onLightTextAccentSecondary = '#FF4791FF';
/** Второй акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradientSecondary = 'linear-gradient(75.33deg, #FF94C9 0%, #F00563 100%)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = '#171717';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = '#4E4E4E';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = '#898989';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraph = '#232323';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = '#FFFFFF';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = '#171717';
/** Вторичный непрозрачный фон поверхности */
exports.surfaceSolidSecondary = '#262626FF';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = '#363636';
/** Основной фон для карточек */
exports.surfaceSolidCard = '#171717';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = '#FFFFFF0F';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = '#119AF0FF';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = '#24B23EFF';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarning = '#FF7024FF';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = '#FF293EFF';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'rgba(0, 0, 0, 0)';
/** Второй акцентный фон поверхности/контрола */
exports.surfaceAccentSecondary = '#FF4791FF';
/** Второй акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradientSecondary = 'linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = '#FFFFFF';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = '#171717';
/** Вторичный непрозрачный фон поверхности на темном фоне */
exports.onDarkSurfaceSolidSecondary = '#262626FF';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = '#363636';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = '#171717';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'rgba(255, 255, 255, 0.12)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = '#119AF0FF';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = '#24B23EFF';
/** Цвет фона поверхности предупреждение на темном фоне */
exports.onDarkSurfaceWarning = '#FF7024FF';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = '#FF293EFF';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClear = 'rgba(0, 0, 0, 0)';
/** Второй акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentSecondary = '#FF4791FF';
/** Второй акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradientSecondary = 'linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%)';
/** Непрозрачный фон поверхности по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = '#171717FF';
/** Основной непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidPrimary = '#F5F5F5FF';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidSecondary = '#ECECECFF';
/** Третичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidTertiary = '#DDDDDDFF';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = '#FFFFFFFF';
/** Основной прозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceTransparentPrimary = '#08080808';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'rgba(8, 8, 8, 0.06)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'rgba(8, 8, 8, 0.12)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = '#FFFFFFFF';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = '#119AF0FF';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'linear-gradient(75.33deg, #12C2F3 0%, #2B8CED 100%)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = '#1A9E32FF';
/** Цвет фона поверхности предупреждение на светлом фоне */
exports.onLightSurfaceWarning = '#FA5F05FF';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = '#FF293EFF';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClear = 'rgba(0, 0, 0, 0)';
/** Второй акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentSecondary = '#FF4791FF';
/** Второй акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradientSecondary = 'linear-gradient(75.33deg, #FC95DA 0%, #FF1F78 100%)';
/** Основной фон */
exports.backgroundPrimary = '#080808FF';
/** Основной фон на темном фоне */
exports.darkBackgroundPrimary = '#080808FF';
/** Основной фон на светлом фоне */
exports.lightBackgroundPrimary = '#F9F9F9FF';
/** Инвертированный основной фон */
exports.inverseBackgroundPrimary = '#F9F9F9FF';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = '#F9F9F9FF';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = '#949494FF';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = '#4E4E4EFF';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = '#171717';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = '#4E4E4E';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = '#898989';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = '#171717FF';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'rgba(8, 8, 8, 0.06)';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = '#FFFFFF';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** @deprecated instead use textPrimary */
exports.text = '#F9F9F9FF';
/** @deprecated instead use textPrimary */
exports.primary = '#F9F9F9FF';
/** @deprecated instead use textSecondary */
exports.secondary = '#949494FF';
/** @deprecated instead use textTertiary */
exports.tertiary = '#4E4E4EFF';
/** @deprecated instead use textParagraph */
exports.paragraph = '#C7C7C7FF';
/** @deprecated instead use backgroundPrimary */
exports.background = '#080808FF';
/** @deprecated instead use textAccent */
exports.accent = '#52BAFFFF';
/** @deprecated instead use textPositive */
exports.success = '#24B23EFF';
/** @deprecated instead use textWarning */
exports.warning = '#FF7024FF';
/** @deprecated instead use textNegative */
exports.critical = '#FF3D51FF';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'rgba(255, 255, 255, 0.06)';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'rgba(255, 255, 255, 0.12)';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'rgba(255, 255, 255, 0.2)';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = '#171717';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = '#262626FF';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = '#363636';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = '#FFFFFF0F';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'rgba(255, 255, 255, 0.12)';
/** @deprecated instead use textAccent */
exports.buttonAccent = '#52BAFFFF';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = '#24B23EFF';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = '#FF7024FF';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = '#FF293EFF';
