// @salutejs/plasma-tokens/brands/sbermarket_selgros/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonCritical = exports.buttonAccent = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.critical = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.backgroundOverlay = exports.backgroundPrimary = exports.bannersDefaultBannerBlue = exports.bannersDefaultBannerMalachite = exports.bannersDefaultBannerGreen = exports.bannersDefaultBannerGold = exports.bannersDefaultBannerRed = exports.onLightSurfaceAdditional = exports.onLightSurfaceAccentBlue = exports.onLightSurfaceAccentGold = exports.onLightSurfaceAccentDiscount = exports.onLightSurfaceAccentB2B = exports.onLightSurfaceNegative = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccent = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondary = exports.onDarkSurfaceAdditional = exports.onDarkSurfaceAccentBlue = exports.onDarkSurfaceAccentGold = exports.onDarkSurfaceAccentDiscount = exports.onDarkSurfaceAccentB2B = exports.onDarkSurfaceNegative = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccent = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondary = exports.surfaceAdditional = exports.surfaceAccentBlue = exports.surfaceAccentGold = exports.surfaceAccentDiscount = exports.surfaceAccentB2B = exports.surfaceNegative = exports.surfaceAccentGradient = exports.surfaceAccent = exports.surfaceSolidCard = exports.surfaceSolidTertiary = exports.surfaceSolidSecondary = exports.inverseTextPrimary = exports.onLightTextAccentB2B = exports.onLightTextAccentGold = exports.onLightTextAccentDiscount = exports.onLightTextNegative = exports.onLightTextAccent = exports.onLightTextParagraph = exports.onLightTextTertiary = exports.onLightTextSecondary = exports.onLightTextPrimary = exports.onDarkTextAccentB2B = exports.onDarkTextAccentGold = exports.onDarkTextAccentDiscount = exports.onDarkTextNegative = exports.onDarkTextAccent = exports.onDarkTextParagraph = exports.onDarkTextTertiary = exports.onDarkTextSecondary = exports.onDarkTextPrimary = exports.textAccentB2B = exports.textAccentGold = exports.textAccentDiscount = exports.textNegative = exports.textAccent = exports.textParagraph = exports.textTertiary = exports.textSecondary = exports.textPrimary = void 0;
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #F2F5F7)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #A8B4BD)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #7D868C)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, #BEC9D1)';
/** Акцентный цвет */
exports.textAccent = 'var(--text-accent, #E40421)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF2E43)';
/** Дополнительный акцент */
exports.textAccentDiscount = 'var(--text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.textAccentGold = 'var(--text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
exports.textAccentB2B = 'var(--text-accent-b2-b, #04C6C9)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, rgba(242, 245, 247, 0.96))';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, rgba(242, 245, 247, 0.64))';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, rgba(242, 245, 247, 0.4))';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, rgba(242, 245, 247, 0.8))';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = 'var(--on-dark-text-accent, #E40421)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF2E43)';
/** Дополнительный акцент */
exports.onDarkTextAccentDiscount = 'var(--on-dark-text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.onDarkTextAccentGold = 'var(--on-dark-text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
exports.onDarkTextAccentB2B = 'var(--on-dark-text-accent-b2-b, #04C6C9)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, #24282B)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, #8C959C)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #BEC9D1)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, #24282B)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = 'var(--on-light-text-accent, #E40421)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Дополнительный акцент */
exports.onLightTextAccentDiscount = 'var(--on-light-text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.onLightTextAccentGold = 'var(--on-light-text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
exports.onLightTextAccentB2B = 'var(--on-light-text-accent-b2-b, #0C8688)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, #24282B)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #31373B)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #15181A)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = 'var(--surface-accent, #E40421)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF2E43)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentB2B = 'var(--surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
exports.surfaceAccentDiscount = 'var(--surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.surfaceAccentGold = 'var(--surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
exports.surfaceAccentBlue = 'var(--surface-accent-blue, #5993FF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.surfaceAdditional = 'var(--surface-additional, #494F54)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #31373B)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #15181A)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #E40421)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF2E43)';
/** Акцентный фон поверхности/контрола */
exports.onDarkSurfaceAccentB2B = 'var(--on-dark-surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
exports.onDarkSurfaceAccentDiscount = 'var(--on-dark-surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentGold = 'var(--on-dark-surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentBlue = 'var(--on-dark-surface-accent-blue, #5993FF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onDarkSurfaceAdditional = 'var(--on-dark-surface-additional, #494F54)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #F2F5F7)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #E4EBF0)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = 'var(--on-light-surface-accent, #E40421)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #F31B31)';
/** Акцентный фон поверхности/контрола */
exports.onLightSurfaceAccentB2B = 'var(--on-light-surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
exports.onLightSurfaceAccentDiscount = 'var(--on-light-surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.onLightSurfaceAccentGold = 'var(--on-light-surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
exports.onLightSurfaceAccentBlue = 'var(--on-light-surface-accent-blue, #3F81FD)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onLightSurfaceAdditional = 'var(--on-light-surface-additional, #D1DDE5)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerRed = 'var(--banners-default-banner-red, #2E090D)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerGold = 'var(--banners-default-banner-gold, #1F1903)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerGreen = 'var(--banners-default-banner-green, #071F0C)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerMalachite = 'var(--banners-default-banner-malachite, #0A1C1C)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerBlue = 'var(--banners-default-banner-blue, #0A162B)';
/** Основной фон */
exports.backgroundPrimary = 'var(--background-primary, #08090A)';
/** Вторичный фон */
exports.backgroundOverlay = 'var(--background-overlay, rgba(21, 24, 26, 0.56))';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, rgba(242, 245, 247, 0.96))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, rgba(242, 245, 247, 0.64))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, rgba(242, 245, 247, 0.4))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, #24282B)';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, #8C959C)';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, #BEC9D1)';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, #F2F5F7)';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, #F2F5F7)';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, #A8B4BD)';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, #7D868C)';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, #BEC9D1)';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, #08090A)';
/** @deprecated instead use textAccent */
exports.accent = 'var(--plasma-colors-accent, #E40421)';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, #FF2E43)';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, #24282B)';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, #31373B)';
/** @deprecated instead use textAccent */
exports.buttonAccent = 'var(--plasma-colors-button-accent, #E40421)';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, #FF2E43)';
