// @salutejs/plasma-tokens/brands/sbermarket_metro/values.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonCritical = exports.buttonAccent = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.critical = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.backgroundOverlay = exports.backgroundPrimary = exports.bannersDefaultBannerBlue = exports.bannersDefaultBannerMalachite = exports.bannersDefaultBannerGreen = exports.bannersDefaultBannerGold = exports.bannersDefaultBannerRed = exports.onLightSurfaceAdditional = exports.onLightSurfaceAccentBlue = exports.onLightSurfaceAccentGold = exports.onLightSurfaceAccentDiscount = exports.onLightSurfaceAccentB2B = exports.onLightSurfaceNegative = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccent = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondary = exports.onDarkSurfaceAdditional = exports.onDarkSurfaceAccentBlue = exports.onDarkSurfaceAccentGold = exports.onDarkSurfaceAccentDiscount = exports.onDarkSurfaceAccentB2B = exports.onDarkSurfaceNegative = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccent = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondary = exports.surfaceAdditional = exports.surfaceAccentBlue = exports.surfaceAccentGold = exports.surfaceAccentDiscount = exports.surfaceAccentB2B = exports.surfaceNegative = exports.surfaceAccentGradient = exports.surfaceAccent = exports.surfaceSolidCard = exports.surfaceSolidTertiary = exports.surfaceSolidSecondary = exports.inverseTextPrimary = exports.onLightTextAccentB2B = exports.onLightTextAccentGold = exports.onLightTextAccentDiscount = exports.onLightTextNegative = exports.onLightTextAccent = exports.onLightTextParagraph = exports.onLightTextTertiary = exports.onLightTextSecondary = exports.onLightTextPrimary = exports.onDarkTextAccentB2B = exports.onDarkTextAccentGold = exports.onDarkTextAccentDiscount = exports.onDarkTextNegative = exports.onDarkTextAccent = exports.onDarkTextParagraph = exports.onDarkTextTertiary = exports.onDarkTextSecondary = exports.onDarkTextPrimary = exports.textAccentB2B = exports.textAccentGold = exports.textAccentDiscount = exports.textNegative = exports.textAccent = exports.textParagraph = exports.textTertiary = exports.textSecondary = exports.textPrimary = void 0;
/** Основной цвет текста */
exports.textPrimary = '#F2F5F7';
/** Вторичный цвет текста */
exports.textSecondary = '#A8B4BD';
/** Третичный цвет текста */
exports.textTertiary = '#7D868C';
/** Сплошной наборный текст */
exports.textParagraph = '#BEC9D1';
/** Акцентный цвет */
exports.textAccent = '#0064FE';
/** Цвет ошибки */
exports.textNegative = '#FF2E43';
/** Дополнительный акцент */
exports.textAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
exports.textAccentGold = '#FFD83D';
/** Акцентный цвет бренда */
exports.textAccentB2B = '#04C6C9';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'rgba(242, 245, 247, 0.96)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'rgba(242, 245, 247, 0.64)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'rgba(242, 245, 247, 0.4)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'rgba(242, 245, 247, 0.8)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = '#0064FE';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = '#FF2E43';
/** Дополнительный акцент */
exports.onDarkTextAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
exports.onDarkTextAccentGold = '#FFD83D';
/** Акцентный цвет бренда */
exports.onDarkTextAccentB2B = '#04C6C9';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = '#24282B';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = '#8C959C';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = '#BEC9D1';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = '#24282B';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = '#0064FE';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = '#F31B31';
/** Дополнительный акцент */
exports.onLightTextAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
exports.onLightTextAccentGold = '#FFD83D';
/** Акцентный цвет бренда */
exports.onLightTextAccentB2B = '#0C8688';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = '#24282B';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = '#24282B';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = '#31373B';
/** Основной фон для карточек */
exports.surfaceSolidCard = '#15181A';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = '#0064FE';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = '#FF2E43';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentB2B = '#0DA8AB';
/** Дополнительный акцент для обозначения акций */
exports.surfaceAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
exports.surfaceAccentGold = '#FFD83D';
/** Дополнительный акцент */
exports.surfaceAccentBlue = '#5993FF';
/** Пунктирные дивайдеры и неактивные контролы */
exports.surfaceAdditional = '#494F54';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondary = '#24282B';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = '#31373B';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = '#15181A';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = '#0064FE';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = '#FF2E43';
/** Акцентный фон поверхности/контрола */
exports.onDarkSurfaceAccentB2B = '#0DA8AB';
/** Дополнительный акцент для обозначения акций */
exports.onDarkSurfaceAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentGold = '#FFD83D';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentBlue = '#5993FF';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onDarkSurfaceAdditional = '#494F54';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondary = '#F2F5F7';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiary = '#E4EBF0';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = '#FFFFFF';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = '#0064FE';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = '#F31B31';
/** Акцентный фон поверхности/контрола */
exports.onLightSurfaceAccentB2B = '#0DA8AB';
/** Дополнительный акцент для обозначения акций */
exports.onLightSurfaceAccentDiscount = '#FF1F78';
/** Дополнительный акцент */
exports.onLightSurfaceAccentGold = '#FFD83D';
/** Дополнительный акцент */
exports.onLightSurfaceAccentBlue = '#3F81FD';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onLightSurfaceAdditional = '#D1DDE5';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerRed = '#2E090D';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerGold = '#1F1903';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerGreen = '#071F0C';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerMalachite = '#0A1C1C';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerBlue = '#0A162B';
/** Основной фон */
exports.backgroundPrimary = '#08090A';
/** Вторичный фон */
exports.backgroundOverlay = 'rgba(21, 24, 26, 0.56)';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'rgba(242, 245, 247, 0.96)';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'rgba(242, 245, 247, 0.64)';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'rgba(242, 245, 247, 0.4)';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = '#24282B';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = '#8C959C';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = '#BEC9D1';
/** @deprecated instead use textPrimary */
exports.text = '#F2F5F7';
/** @deprecated instead use textPrimary */
exports.primary = '#F2F5F7';
/** @deprecated instead use textSecondary */
exports.secondary = '#A8B4BD';
/** @deprecated instead use textTertiary */
exports.tertiary = '#7D868C';
/** @deprecated instead use textParagraph */
exports.paragraph = '#BEC9D1';
/** @deprecated instead use backgroundPrimary */
exports.background = '#08090A';
/** @deprecated instead use textAccent */
exports.accent = '#0064FE';
/** @deprecated instead use textNegative */
exports.critical = '#FF2E43';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = '#24282B';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = '#31373B';
/** @deprecated instead use textAccent */
exports.buttonAccent = '#0064FE';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = '#FF2E43';
