// @salutejs/plasma-ui/components/Card/Card.examples.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GalleryCard = exports.MusicCard = exports.ProductCard = void 0;

var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _plasmaTokens = /*#__PURE__*/require("@salutejs/plasma-tokens");

var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");

var _Typography = /*#__PURE__*/require("../Typography");

var _Card = /*#__PURE__*/require("./Card");

var _CardBody = /*#__PURE__*/require("./CardBody");

var _CardContent = /*#__PURE__*/require("./CardContent");

var _CardMedia = /*#__PURE__*/require("./CardMedia");

var _excluded = ["title", "subtitle", "imageSrc"],
    _excluded2 = ["title", "subtitle", "roundness", "imageSrc", "imageRatio", "imageCustomRatio", "textAlign"],
    _excluded3 = ["title", "subtitle", "imageSrc", "imageRatio", "imageCustomRatio", "textAlign"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var applyTextAlign = function applyTextAlign(_ref) {
  var textAlign = _ref.textAlign;
  return textAlign && (0, _styledComponents.css)(["text-align:", ";"], textAlign);
};

var StyledProductIcon = /*#__PURE__*/(0, _styledComponents["default"])(_CardMedia.CardMedia).withConfig({
  componentId: "plasma-ui__sc-3b5xy8-0"
})(["width:3rem;height:3rem;padding:0;border-radius:1rem;"]);
var StyledItemDescr = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote1).withConfig({
  componentId: "plasma-ui__sc-3b5xy8-1"
})(["", " ", " margin-top:0.25rem;color:", ";", " + &{margin-top:1rem;}"], _plasmaCore.applyMaxLines, applyTextAlign, _plasmaTokens.secondary, StyledProductIcon);
var StyledItemTitle = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body1).withConfig({
  componentId: "plasma-ui__sc-3b5xy8-2"
})(["", " ", " margin-top:0.75rem;color:", ";", " + &{margin-top:0.25rem;}"], _plasmaCore.applyMaxLines, applyTextAlign, _plasmaTokens.primary, StyledItemDescr);

var ProductCard = function ProductCard(_ref2) {
  var title = _ref2.title,
      subtitle = _ref2.subtitle,
      imageSrc = _ref2.imageSrc,
      rest = _objectWithoutProperties(_ref2, _excluded);

  return /*#__PURE__*/_react["default"].createElement(_Card.Card, rest, /*#__PURE__*/_react["default"].createElement(_CardBody.CardBody, null, /*#__PURE__*/_react["default"].createElement(_CardContent.CardContent, null, /*#__PURE__*/_react["default"].createElement(StyledProductIcon, {
    src: imageSrc
  }), subtitle && /*#__PURE__*/_react["default"].createElement(StyledItemDescr, {
    maxLines: 1
  }, subtitle), /*#__PURE__*/_react["default"].createElement(StyledItemTitle, {
    maxLines: 1
  }, title))));
};

exports.ProductCard = ProductCard;

var MusicCard = function MusicCard(_ref3) {
  var title = _ref3.title,
      subtitle = _ref3.subtitle,
      _ref3$roundness = _ref3.roundness,
      roundness = _ref3$roundness === void 0 ? 12 : _ref3$roundness,
      imageSrc = _ref3.imageSrc,
      imageRatio = _ref3.imageRatio,
      imageCustomRatio = _ref3.imageCustomRatio,
      textAlign = _ref3.textAlign,
      rest = _objectWithoutProperties(_ref3, _excluded2);

  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Card.Card, _extends({
    roundness: roundness
  }, rest), /*#__PURE__*/_react["default"].createElement(_CardBody.CardBody, null, /*#__PURE__*/_react["default"].createElement(_CardMedia.CardMedia, {
    src: imageSrc,
    ratio: imageRatio,
    customRatio: imageCustomRatio
  }))), /*#__PURE__*/_react["default"].createElement(StyledItemTitle, {
    maxLines: 1,
    textAlign: textAlign
  }, title), subtitle && /*#__PURE__*/_react["default"].createElement(StyledItemDescr, {
    maxLines: 1,
    textAlign: textAlign
  }, subtitle));
};

exports.MusicCard = MusicCard;

var GalleryCard = function GalleryCard(_ref4) {
  var title = _ref4.title,
      subtitle = _ref4.subtitle,
      imageSrc = _ref4.imageSrc,
      imageRatio = _ref4.imageRatio,
      imageCustomRatio = _ref4.imageCustomRatio,
      textAlign = _ref4.textAlign,
      rest = _objectWithoutProperties(_ref4, _excluded3);

  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Card.Card, rest, /*#__PURE__*/_react["default"].createElement(_CardBody.CardBody, null, /*#__PURE__*/_react["default"].createElement(_CardMedia.CardMedia, {
    src: imageSrc,
    ratio: imageRatio,
    customRatio: imageCustomRatio
  }), /*#__PURE__*/_react["default"].createElement(_CardContent.CardContent, {
    cover: true
  }, /*#__PURE__*/_react["default"].createElement(StyledItemTitle, {
    maxLines: 1,
    textAlign: textAlign
  }, title), subtitle && /*#__PURE__*/_react["default"].createElement(StyledItemDescr, {
    maxLines: 1,
    textAlign: textAlign
  }, subtitle)))));
};

exports.GalleryCard = GalleryCard;
