// @salutejs/plasma-ui/components/Radiobox/Radiobox.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Radiobox = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");

var _plasmaTokens = /*#__PURE__*/require("@salutejs/plasma-tokens");

var _Checkbox = /*#__PURE__*/require("../Checkbox/Checkbox");

var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");

var _StyledEllipse;

var _excluded = ["id", "label", "description", "disabled", "focused", "scaleOnInteraction", "style", "className"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

(0, _collectPackageInfo.collectPackageInfo)('Radiobox/Radiobox');
var StyledTrigger = /*#__PURE__*/(0, _styledComponents["default"])(_Checkbox.StyledTrigger).withConfig({
  componentId: "plasma-ui__sc-1ol40bw-0"
})(["--plasma-trigger-border-radius:1.25rem;--plasma-trigger-outline-radius:1.375rem;"]); // TODO: https://github.com/salute-developers/plasma/issues/232

var StyledEllipse = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-1ol40bw-1"
})(["position:absolute;top:0;left:0;right:0;bottom:0;margin:auto;width:0.625rem;height:0.625rem;border-radius:0.625rem;background-color:", ";transition:", ";transform:scale(0);input:checked + label ", " &{transform:scale(1);}"], _plasmaTokens.white, function (_ref) {
  var theme = _ref.theme;
  return theme.lowPerformance ? 'unset' : 'transform 0.3s ease-in-out';
}, StyledTrigger);
/**
 * Переключатель, или *радиокнопка*.
 */
// eslint-disable-next-line prefer-arrow-callback


var Radiobox = /*#__PURE__*/(0, _react.forwardRef)(function Radiobox(_ref2, ref) {
  var id = _ref2.id,
      label = _ref2.label,
      description = _ref2.description,
      disabled = _ref2.disabled,
      focused = _ref2.focused,
      scaleOnInteraction = _ref2.scaleOnInteraction,
      style = _ref2.style,
      className = _ref2.className,
      rest = _objectWithoutProperties(_ref2, _excluded);

  var uniqId = (0, _plasmaCore.useUniqId)();
  var uniqLabelId = (0, _plasmaCore.useUniqId)();
  var uniqDescriptionId = (0, _plasmaCore.useUniqId)();
  var radioboxId = id || uniqId;
  return /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledRoot, {
    $disabled: disabled,
    style: style,
    className: className,
    tabIndex: -1
  }, /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledInput, _extends({
    "aria-labelledby": uniqLabelId,
    "aria-describedby": uniqDescriptionId,
    id: radioboxId,
    ref: ref,
    type: "radio",
    disabled: disabled
  }, rest)), /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxContentWrapper, {
    htmlFor: radioboxId
  }, /*#__PURE__*/_react["default"].createElement(StyledTrigger, {
    $focused: focused,
    $scaleOnInteraction: scaleOnInteraction
  }, _StyledEllipse || (_StyledEllipse = /*#__PURE__*/_react["default"].createElement(StyledEllipse, null))), label && /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledContent, null, /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxLabel, {
    as: "span",
    id: uniqLabelId
  }, label), description && /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxDescription, {
    mt: 4,
    id: uniqDescriptionId
  }, description))));
});
exports.Radiobox = Radiobox;
