// @salutejs/plasma-ui/components/Pickers/PickerItem.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PickerItem = exports.StyledWhiteText = exports.StyledPickerItem = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _plasmaTokens = /*#__PURE__*/require("@salutejs/plasma-tokens");

var _utils = /*#__PURE__*/require("./utils");

var _excluded = ["size", "item", "index", "activeIndex", "noScrollBehavior", "onItemClick", "autofocus", "disabled", "disableScrollSnapAlign"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var sizes = {
  l: /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, _plasmaTokens.display2), {}, {
    padding: '0.5rem 0'
  }),
  s: /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, _plasmaTokens.headline1), {}, {
    fontWeight: 600,
    padding: 0
  }),
  xs: /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, _plasmaTokens.headline2), {}, {
    padding: 0
  })
};

var StyledPickerItem = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-158n193-0"
})(["position:relative;box-sizing:border-box;overflow:hidden;display:flex;align-items:center;text-align:center;width:100%;color:", ";cursor:pointer;user-select:none;", " &:focus{outline:0 none;background:none;}", " ", ""], _plasmaTokens.tertiary, function (_ref) {
  var $noScrollBehavior = _ref.$noScrollBehavior,
      $disableScrollSnapAlign = _ref.$disableScrollSnapAlign;
  return !$noScrollBehavior && (0, _styledComponents.css)(["scroll-snap-align:", ";"], $disableScrollSnapAlign ? 'initial' : 'center');
}, function (_ref2) {
  var isSnapAlwaysStop = _ref2.isSnapAlwaysStop;
  return isSnapAlwaysStop && (0, _styledComponents.css)(["scroll-snap-stop:always;"]);
}, function (_ref3) {
  var $size = _ref3.$size;
  return sizes[$size];
}); // TODO: https://github.com/salute-developers/plasma/issues/232


exports.StyledPickerItem = StyledPickerItem;

var StyledTransformable = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-158n193-1"
})(["width:100%;", " flex-direction:column;", ""], function (_ref4) {
  var $disableScrollSnapAlign = _ref4.$disableScrollSnapAlign;
  return !$disableScrollSnapAlign && (0, _styledComponents.css)(["height:100%;"]);
}, function (_ref5) {
  var $noScrollBehavior = _ref5.$noScrollBehavior;
  return !$noScrollBehavior && (0, _styledComponents.css)(["transition:", ";transform:translate3d(0,0,0);"], function (_ref6) {
    var theme = _ref6.theme;
    return theme.lowPerformance ? 'unset' : 'transform 0.1s ease';
  });
});

var StyledText = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma-ui__sc-158n193-2"
})(["transform:translate3d(0,0,0);"]);

var StyledWhiteText = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-158n193-3"
})(["position:absolute;top:0;left:0;right:0;bottom:0;color:", ";"], _plasmaTokens.secondary);

exports.StyledWhiteText = StyledWhiteText;

var PickerItem = function PickerItem(_ref7) {
  var _ref7$size = _ref7.size,
      size = _ref7$size === void 0 ? 's' : _ref7$size,
      item = _ref7.item,
      index = _ref7.index,
      activeIndex = _ref7.activeIndex,
      noScrollBehavior = _ref7.noScrollBehavior,
      onItemClick = _ref7.onItemClick,
      autofocus = _ref7.autofocus,
      disabled = _ref7.disabled,
      disableScrollSnapAlign = _ref7.disableScrollSnapAlign,
      rest = _objectWithoutProperties(_ref7, _excluded);

  var itemRef = _react["default"].useRef(null);
  /*
   * Выведем стили еще до того, как отработает коллбек стилей.
   * Тут важно, что для `slot` идут целочисленные значения.
   */


  var styles = _react["default"].useMemo(function () {
    return (0, _utils.getStyles)(index - activeIndex, size);
  }, [index, activeIndex, size]);

  var onClick = _react["default"].useCallback(function () {
    if (disabled) {
      return;
    }

    onItemClick === null || onItemClick === void 0 ? void 0 : onItemClick(item);
  }, [item, disabled]);

  (0, _react.useEffect)(function () {
    if (autofocus && itemRef.current) {
      itemRef.current.focus();
    }
  }, [autofocus]);
  return /*#__PURE__*/_react["default"].createElement(StyledPickerItem, _extends({
    $noScrollBehavior: noScrollBehavior,
    $disableScrollSnapAlign: disableScrollSnapAlign,
    ref: itemRef,
    $size: size,
    onClick: onClick
  }, rest), /*#__PURE__*/_react["default"].createElement(StyledTransformable, {
    $noScrollBehavior: noScrollBehavior,
    $disableScrollSnapAlign: disableScrollSnapAlign,
    $size: size,
    style: styles.wrapper
  }, /*#__PURE__*/_react["default"].createElement(StyledText, {
    style: styles.text
  }, item.label), /*#__PURE__*/_react["default"].createElement(StyledWhiteText, {
    style: styles.whiteText,
    "aria-hidden": "true"
  }, item.label)));
};

exports.PickerItem = PickerItem;
