// @salutejs/plasma-core/components/Typography/Paragraph.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.P2 = exports.P = exports.P1 = exports.ParagraphText2 = exports.ParagraphText1 = void 0;

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _tokens = /*#__PURE__*/require("../../tokens");

var _mixins = /*#__PURE__*/require("../../mixins");

var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

(0, _collectPackageInfo.collectPackageInfo)('Typography/Paragraph');

var ParagraphText1 = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-137f45k-0"
})(["", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.paragraph1);

exports.ParagraphText1 = ParagraphText1;

var ParagraphText2 = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-137f45k-1"
})(["", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.paragraph2);

exports.ParagraphText2 = ParagraphText2;

var P1 = /*#__PURE__*/_styledComponents["default"].p.withConfig({
  componentId: "plasma__sc-137f45k-2"
})(["", " ", " ", " margin:0;"], _mixins.applyHyphens, _mixins.applySpacing, _tokens.paragraph1);

exports.P1 = P1;
var P = P1;
exports.P = P;

var P2 = /*#__PURE__*/_styledComponents["default"].p.withConfig({
  componentId: "plasma__sc-137f45k-3"
})(["", " ", " ", " margin:0;"], _mixins.applyHyphens, _mixins.applySpacing, _tokens.paragraph2);

exports.P2 = P2;
