// @salutejs/plasma-ui/components/Checkbox/Checkbox.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Checkbox = exports.StyledContent = exports.StyledTrigger = exports.StyledInput = exports.StyledRoot = exports.syntheticFocus = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");

var _plasmaTokens = /*#__PURE__*/require("@salutejs/plasma-tokens");

var _utils = /*#__PURE__*/require("../../utils");

var _IconDone = /*#__PURE__*/require("./IconDone");

var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");

var _StyledMark;

var _excluded = ["id", "label", "description", "disabled", "focused", "scaleOnInteraction", "style", "className"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

(0, _collectPackageInfo.collectPackageInfo)('Checkbox/Checkbox');

var syntheticFocus = function syntheticFocus(ruleset, focused) {
  return (0, _styledComponents.css)(["input[data-focus-visible-added] + label &{box-shadow:none;outline:none;", "}", ""], ruleset, focused && (0, _styledComponents.css)(["box-shadow:none;", ""], ruleset));
};

exports.syntheticFocus = syntheticFocus;
var StyledRoot = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxRoot).withConfig({
  componentId: "plasma-ui__sc-i8k5cq-0"
})(["", ";& + &{margin-top:1.25rem;}"], _plasmaCore.applyDisabled);
exports.StyledRoot = StyledRoot;
var StyledInput = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxInput).attrs(_utils.spatnavClassNameAttrs).withConfig({
  componentId: "plasma-ui__sc-i8k5cq-1"
})(["appearance:none;width:100%;height:100%;opacity:0;"]); // TODO: https://github.com/salute-developers/plasma/issues/232

exports.StyledInput = StyledInput;
var StyledTrigger = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxTrigger).withConfig({
  componentId: "plasma-ui__sc-i8k5cq-2"
})(["--plasma-trigger-border-radius:0.25rem;--plasma-trigger-outline-radius:0.375rem;width:1.25rem;height:1.25rem;flex:0 0 1.25rem;background:", ";box-shadow:inset 0 0 0 0.125rem ", ";color:", ";input:checked + label &{background:", ";box-shadow:inset 0 0 0 0.125rem ", ";}", " border-radius:var(--plasma-trigger-border-radius);", ""], _plasmaTokens.transparent, _plasmaTokens.secondary, _plasmaTokens.white, _plasmaTokens.accent, _plasmaTokens.accent, function (_ref) {
  var $scaleOnInteraction = _ref.$scaleOnInteraction;
  return $scaleOnInteraction && (0, _styledComponents.css)(["&:hover{transform:scale(1.04);}&:active{transform:scale(0.96);}"]);
}, function (_ref2) {
  var $focused = _ref2.$focused,
      theme = _ref2.theme;
  return (0, _styledComponents.css)(["", ""], (0, _plasmaCore.addFocus)({
    synthesizeFocus: syntheticFocus,
    focused: $focused,
    lowPerformance: theme.lowPerformance,
    outlineRadius: 'var(--plasma-trigger-outline-radius)'
  }));
});
exports.StyledTrigger = StyledTrigger;
var StyledMark = /*#__PURE__*/(0, _styledComponents["default"])(_IconDone.IconDone).withConfig({
  componentId: "plasma-ui__sc-i8k5cq-3"
})(["position:absolute;top:0;left:0;right:0;bottom:0;margin:auto;width:1rem;height:1rem;transition:", ";transform:scale(0);input:checked + label ", " &{transform:scale(1);}"], function (_ref3) {
  var theme = _ref3.theme;
  return theme.lowPerformance ? 'unset' : 'transform 0.1s ease-in-out';
}, StyledTrigger);
var StyledContent = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxContent).withConfig({
  componentId: "plasma-ui__sc-i8k5cq-4"
})(["width:calc(100% - 2rem);"]);
/**
 * Флажок или *чекбокс*. Позволяет пользователю управлять параметром с двумя состояниями — ☑ включено и ☐ отключено.
 */
// eslint-disable-next-line prefer-arrow-callback

exports.StyledContent = StyledContent;
var Checkbox = /*#__PURE__*/(0, _react.forwardRef)(function Checkbox(_ref4, ref) {
  var id = _ref4.id,
      label = _ref4.label,
      description = _ref4.description,
      disabled = _ref4.disabled,
      focused = _ref4.focused,
      scaleOnInteraction = _ref4.scaleOnInteraction,
      style = _ref4.style,
      className = _ref4.className,
      rest = _objectWithoutProperties(_ref4, _excluded);

  var uniqId = (0, _plasmaCore.useUniqId)();
  var uniqLabelId = (0, _plasmaCore.useUniqId)();
  var uniqDescriptionId = (0, _plasmaCore.useUniqId)();
  var checkboxId = id || uniqId;
  return /*#__PURE__*/_react["default"].createElement(StyledRoot, {
    $disabled: disabled,
    style: style,
    className: className,
    tabIndex: -1
  }, /*#__PURE__*/_react["default"].createElement(StyledInput, _extends({
    "aria-labelledby": uniqLabelId,
    "aria-describedby": uniqDescriptionId,
    id: checkboxId,
    ref: ref,
    type: "checkbox",
    disabled: disabled
  }, rest)), /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxContentWrapper, {
    htmlFor: checkboxId
  }, /*#__PURE__*/_react["default"].createElement(StyledTrigger, {
    $focused: focused,
    $scaleOnInteraction: scaleOnInteraction
  }, _StyledMark || (_StyledMark = /*#__PURE__*/_react["default"].createElement(StyledMark, {
    color: "inherit",
    size: "xs"
  }))), label && /*#__PURE__*/_react["default"].createElement(StyledContent, null, /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxLabel, {
    as: "span",
    id: uniqLabelId
  }, label), description && /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxDescription, {
    mt: 4,
    id: uniqDescriptionId
  }, description))));
});
exports.Checkbox = Checkbox;
